
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>10. Command-line interface &#8212; ProPhyle 0.3.1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/sphinx13.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.3.1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="search" type="application/opensearchdescription+xml"
          title="Search within ProPhyle 0.3.1.0 documentation"
          href="_static/opensearch.xml"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="9. File formats" href="formats.html" />
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:300,400,700'
          rel='stylesheet' type='text/css' />
 
    <style type="text/css">
      table.right { float: right; margin-left: 20px; }
      table.right td { border: 1px solid #ccc; }
      
    </style>
    <script type="text/javascript">
      // intelligent scrolling of the sidebar content
      $(window).scroll(function() {
        var sb = $('.sphinxsidebarwrapper');
        var win = $(window);
        var sbh = sb.height();
        var offset = $('.sphinxsidebar').position()['top'];
        var wintop = win.scrollTop();
        var winbot = wintop + win.innerHeight();
        var curtop = sb.position()['top'];
        var curbot = curtop + sbh;
        // does sidebar fit in window?
        if (sbh < win.innerHeight()) {
          // yes: easy case -- always keep at the top
          sb.css('top', $u.min([$u.max([0, wintop - offset - 10]),
                                $(document).height() - sbh - 200]));
        } else {
          // no: only scroll if top/bottom edge of sidebar is at
          // top/bottom edge of window
          if (curtop > wintop && curbot > winbot) {
            sb.css('top', $u.max([wintop - offset - 10, 0]));
          } else if (curtop < wintop && curbot < winbot) {
            sb.css('top', $u.min([winbot - sbh - offset - 20,
                                  $(document).height() - sbh - 200]));
          }
        }
      });
    </script>

  </head>
  <body>
<div class="pageheader">
  <ul>
    <li><a href="index.html">Home</a></li>
    <li><a href="install.html">Get it</a></li>
    <li><a href="contents.html">Docs</a></li>
    <li><a href="http://github.com/prophyle/prophyle">Extend/Develop</a></li>
  </ul>
  <div>
    <a href="index.html">
      <!--
        <img src="_static/sphinxheader.png" alt="SPHINX" />
      -->
      <div style="color:white;font-size:240%">ProPhyle</div>
      <div style="color:white;font-size:140%">DNA sequence classification</div>
    </a>
  </div>
</div>

    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="formats.html" title="9. File formats"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">ProPhyle home</a>&#160;|</li>
        <li><a href="contents.html">Documentation</a> &#187;</li>
 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="contents.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">10. Command-line interface</a><ul>
<li><a class="reference internal" href="#prophyle-the-main-program">10.1. ProPhyle – the main program</a><ul>
<li><a class="reference internal" href="#prophyle"><code class="docutils literal"><span class="pre">prophyle</span></code></a></li>
<li><a class="reference internal" href="#prophyle-download"><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">download</span></code></a></li>
<li><a class="reference internal" href="#prophyle-index"><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">index</span></code></a></li>
<li><a class="reference internal" href="#prophyle-classify"><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">classify</span></code></a></li>
<li><a class="reference internal" href="#prophyle-analyze"><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">analyze</span></code></a></li>
<li><a class="reference internal" href="#prophyle-compress"><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">compress</span></code></a></li>
<li><a class="reference internal" href="#prophyle-decompress"><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">decompress</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#prophyle-assembler">10.2. <code class="docutils literal"><span class="pre">prophyle_assembler</span></code></a></li>
<li><a class="reference internal" href="#id1">10.3. <code class="docutils literal"><span class="pre">prophyle_index</span></code></a></li>
<li><a class="reference internal" href="#prophyle-assignment">10.4. <code class="docutils literal"><span class="pre">prophyle_assignment</span></code></a></li>
<li><a class="reference internal" href="#prophyle-analyze-py">10.5. <code class="docutils literal"><span class="pre">prophyle_analyze.py</span></code></a></li>
<li><a class="reference internal" href="#prophyle-propagation-makefile-py">10.6. <code class="docutils literal"><span class="pre">prophyle_propagation_makefile.py</span></code></a></li>
<li><a class="reference internal" href="#prophyle-propagation-preprocessing-py">10.7. <code class="docutils literal"><span class="pre">prophyle_propagation_preprocessing.py</span></code></a></li>
<li><a class="reference internal" href="#prophyle-propagation-postprocessing-py">10.8. <code class="docutils literal"><span class="pre">prophyle_propagation_postprocessing.py</span></code></a></li>
<li><a class="reference internal" href="#prophyle-validate-tree-py">10.9. <code class="docutils literal"><span class="pre">prophyle_validate_tree.py</span></code></a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="formats.html"
                        title="previous chapter">9. File formats</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/cli.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="command-line-interface">
<span id="cli"></span><h1>10. Command-line interface<a class="headerlink" href="#command-line-interface" title="Permalink to this headline">¶</a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><a class="reference internal" href="#prophyle-the-main-program" id="id2">ProPhyle – the main program</a><ul>
<li><a class="reference internal" href="#prophyle" id="id3"><code class="docutils literal"><span class="pre">prophyle</span></code></a></li>
<li><a class="reference internal" href="#prophyle-download" id="id4"><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">download</span></code></a></li>
<li><a class="reference internal" href="#prophyle-index" id="id5"><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">index</span></code></a></li>
<li><a class="reference internal" href="#prophyle-classify" id="id6"><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">classify</span></code></a></li>
<li><a class="reference internal" href="#prophyle-analyze" id="id7"><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">analyze</span></code></a></li>
<li><a class="reference internal" href="#prophyle-compress" id="id8"><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">compress</span></code></a></li>
<li><a class="reference internal" href="#prophyle-decompress" id="id9"><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">decompress</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#prophyle-assembler" id="id10"><code class="docutils literal"><span class="pre">prophyle_assembler</span></code></a></li>
<li><a class="reference internal" href="#id1" id="id11"><code class="docutils literal"><span class="pre">prophyle_index</span></code></a></li>
<li><a class="reference internal" href="#prophyle-assignment" id="id12"><code class="docutils literal"><span class="pre">prophyle_assignment</span></code></a></li>
<li><a class="reference internal" href="#prophyle-analyze-py" id="id13"><code class="docutils literal"><span class="pre">prophyle_analyze.py</span></code></a></li>
<li><a class="reference internal" href="#prophyle-propagation-makefile-py" id="id14"><code class="docutils literal"><span class="pre">prophyle_propagation_makefile.py</span></code></a></li>
<li><a class="reference internal" href="#prophyle-propagation-preprocessing-py" id="id15"><code class="docutils literal"><span class="pre">prophyle_propagation_preprocessing.py</span></code></a></li>
<li><a class="reference internal" href="#prophyle-propagation-postprocessing-py" id="id16"><code class="docutils literal"><span class="pre">prophyle_propagation_postprocessing.py</span></code></a></li>
<li><a class="reference internal" href="#prophyle-validate-tree-py" id="id17"><code class="docutils literal"><span class="pre">prophyle_validate_tree.py</span></code></a></li>
</ul>
</div>
<div class="section" id="prophyle-the-main-program">
<h2>10.1. ProPhyle – the main program<a class="headerlink" href="#prophyle-the-main-program" title="Permalink to this headline">¶</a></h2>
<div class="section" id="prophyle">
<h3><code class="docutils literal"><span class="pre">prophyle</span></code><a class="headerlink" href="#prophyle" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle  -h

usage: prophyle.py [-h] [-v]  ...

Program: prophyle (phylogeny-based metagenomic classification)
Version: 0.3.1.0
Authors: Karel Brinda, Kamil Salikhov, Simone Pignotti, Gregory Kucherov
Contact: kbrinda@hsph.harvard.edu

Usage:   prophyle &lt;command&gt; [options]

optional arguments:
  -h, --help     show this help message and exit
  -v, --version  show program&#39;s version number and exit

subcommands:

    download     download a genomic database
    index        build index
    classify     classify reads
    analyze      analyze results (experimental)
    compress     compress a ProPhyle index
    decompress   decompress a compressed ProPhyle index
    compile      compile auxiliary ProPhyle programs
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-download">
<h3><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">download</span></code><a class="headerlink" href="#prophyle-download" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle download -h

usage: prophyle.py download [-h] [-d DIR] [-l STR] [-F] [-c [STR [STR ...]]]
                            &lt;library&gt; [&lt;library&gt; ...]

positional arguments:
  &lt;library&gt;           genomic library [&#39;bacteria&#39;, &#39;viruses&#39;, &#39;plasmids&#39;,
                      &#39;hmp&#39;, &#39;all&#39;]

optional arguments:
  -h, --help          show this help message and exit
  -d DIR              directory for the tree and the sequences [~/prophyle]
  -l STR              log file
  -F                  rewrite library files if they already exist
  -c [STR [STR ...]]  advanced configuration (a JSON dictionary)
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-index">
<h3><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">index</span></code><a class="headerlink" href="#prophyle-index" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle index -h

usage: prophyle.py index [-h] [-g DIR] [-j INT] [-k INT] [-l STR] [-s FLOAT]
                         [-F] [-M] [-P] [-K] [-T] [-A] [-c [STR [STR ...]]]
                         &lt;tree.nw&gt; [&lt;tree.nw&gt; ...] &lt;index.dir&gt;

positional arguments:
  &lt;tree.nw&gt;           phylogenetic tree (in Newick/NHX)
  &lt;index.dir&gt;         index directory (will be created)

optional arguments:
  -h, --help          show this help message and exit
  -g DIR              directory with the library sequences [dir. of the first
                      tree]
  -j INT              number of threads [auto (4)]
  -k INT              k-mer length [31]
  -l STR              log file [&lt;index.dir&gt;/log.txt]
  -s FLOAT            rate of sampling of the tree [no sampling]
  -F                  rewrite index files if they already exist
  -M                  mask repeats/low complexity regions (using DustMasker)
  -P                  do not add prefixes to node names when multiple trees
                      are used
  -K                  skip k-LCP construction (then restarted search only)
  -T                  keep temporary files from k-mer propagation
  -A                  autocomplete tree (names of internal nodes and FASTA
                      paths)
  -c [STR [STR ...]]  advanced configuration (a JSON dictionary)
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-classify">
<h3><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">classify</span></code><a class="headerlink" href="#prophyle-classify" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle classify -h

usage: prophyle.py classify [-h] [-k INT] [-m {h1,c1,h2,c2}] [-f {kraken,sam}]
                            [-l STR] [-P] [-A] [-L] [-X] [-M] [-C] [-K]
                            [-c [STR [STR ...]]]
                            &lt;index.dir&gt; &lt;reads1.fq&gt; [&lt;reads2.fq&gt;]

positional arguments:
  &lt;index.dir&gt;         index directory
  &lt;reads1.fq&gt;         first file with reads in FASTA/FASTQ (- for standard
                      input)
  &lt;reads2.fq&gt;         second file with reads in FASTA/FASTQ

optional arguments:
  -h, --help          show this help message and exit
  -k INT              k-mer length [detect automatically from the index]
  -m {h1,c1,h2,c2}    measure: h1=hit count, c1=coverage, h2=norm.hit count,
                      c2=norm.coverage [h1]
  -f {kraken,sam}     output format [sam]
  -l STR              log file
  -P                  incorporate sequences and qualities into SAM records
  -A                  annotate assignments (using tax. information from NHX)
  -L                  replace read assignments by their LCA
  -X                  replace k-mer matches by their LCA
  -M                  mimic Kraken (equivalent to &quot;-m h1 -f kraken -L -X&quot;)
  -C                  use C++ impl. of the assignment algorithm (experimental)
  -K                  force restarted search mode
  -c [STR [STR ...]]  advanced configuration (a JSON dictionary)
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-analyze">
<h3><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">analyze</span></code><a class="headerlink" href="#prophyle-analyze" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle analyze -h

usage: prophyle.py analyze [-h] [-s [&#39;w&#39;, &#39;u&#39;, &#39;wl&#39;, &#39;ul&#39;]]
                           [-f [&#39;sam&#39;, &#39;bam&#39;, &#39;cram&#39;, &#39;uncompressed_bam&#39;, &#39;kraken&#39;, &#39;histo&#39;]]
                           [-c [STR [STR ...]]]
                           {index_dir, tree.nw} &lt;out.pref&gt; &lt;classified.bam&gt;
                           [&lt;classified.bam&gt; ...]

positional arguments:
  {index_dir, tree.nw}     index directory or phylogenetic tree
  &lt;out.pref&gt;               output prefix
  &lt;classified.bam&gt;         classified reads (use &#39;-&#39; for stdin)

optional arguments:
  -h, --help               show this help message and exit
  -s [&#39;w&#39;, &#39;u&#39;, &#39;wl&#39;, &#39;ul&#39;]
                           statistics to use for the computation of
                           histograms: w (default) =&gt; weighted assignments; u
                           =&gt; unique assignments, non-weighted; wl =&gt; weighted
                           assignments, propagated to leaves; ul =&gt; unique
                           assignments, propagated to leaves.
  -f [&#39;sam&#39;, &#39;bam&#39;, &#39;cram&#39;, &#39;uncompressed_bam&#39;, &#39;kraken&#39;, &#39;histo&#39;]
                           Input format of assignments [auto]
  -c [STR [STR ...]]       advanced configuration (a JSON dictionary)
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-compress">
<h3><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">compress</span></code><a class="headerlink" href="#prophyle-compress" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle compress -h

usage: prophyle.py compress [-h] [-c [STR [STR ...]]]
                            &lt;index.dir&gt; [&lt;archive.tar.gz&gt;]

positional arguments:
  &lt;index.dir&gt;         index directory
  &lt;archive.tar.gz&gt;    output archive [&lt;index.dir&gt;.tar.gz]

optional arguments:
  -h, --help          show this help message and exit
  -c [STR [STR ...]]  advanced configuration (a JSON dictionary)
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-decompress">
<h3><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">decompress</span></code><a class="headerlink" href="#prophyle-decompress" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle decompress -h

usage: prophyle.py decompress [-h] [-K] [-c [STR [STR ...]]]
                              &lt;archive.tar.gz&gt; [&lt;output.dir&gt;]

positional arguments:
  &lt;archive.tar.gz&gt;    output archive
  &lt;output.dir&gt;        output directory [./]

optional arguments:
  -h, --help          show this help message and exit
  -K                  skip k-LCP construction
  -c [STR [STR ...]]  advanced configuration (a JSON dictionary)
</pre></div>
</div>
</div></blockquote>
</div>
</div>
<div class="section" id="prophyle-assembler">
<h2>10.2. <code class="docutils literal"><span class="pre">prophyle_assembler</span></code><a class="headerlink" href="#prophyle-assembler" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_assembler -h


Program:  prophyle_assembler (greedy assembler for ProPhyle)
Contact:  Karel Brinda &lt;karel.brinda@gmail.com&gt;

Usage:    prophyle_assembler [options]

Examples: prophyle_assembler -k 15 -i f1.fa -i f2.fa -x fx.fa
             - compute intersection of f1 and f2
          prophyle_assembler -k 15 -i f1.fa -i f2.fa -x fx.fa -o g1.fa -o g2.fa
             - compute intersection of f1 and f2, and subtract it from them
          prophyle_assembler -k 15 -i f1.fa -o g1.fa
             - re-assemble f1 to g1

Command-line parameters:
 -k INT   K-mer size.
 -i FILE  Input FASTA file (can be used multiple times).
 -o FILE  Output FASTA file (if used, must be used as many times as -i).
 -x FILE  Compute intersection, subtract it, save it.
 -s FILE  Output file with k-mer statistics.
 -S       Silent mode.

Note that &#39;-&#39; can be used for standard input/output.
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="id1">
<h2>10.3. <code class="docutils literal"><span class="pre">prophyle_index</span></code><a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_index -h


Program: prophyle_index (alignment of k-mers)
Contact: Kamil Salikhov &lt;kamil.salikhov@univ-mlv.fr&gt;

Usage:   prophyle_index command [options]

Command: build     construct index
         query     query reads against index
</pre></div>
</div>
<div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_index build -h


Usage:   prophyle_index build &lt;prefix&gt;

Options: -k INT    length of k-mer
         -s        construct k-LCP and SA in parallel
         -i        sampling distance for SA
</pre></div>
</div>
<div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_index query -h


Usage:   prophyle_index query [options] &lt;prefix&gt; &lt;in.fq&gt;

Options: -k INT    length of k-mer
         -u        use k-LCP for querying
         -v        output set of chromosomes for every k-mer
         -p        do not check whether k-mer is on border of two contigs, and show such k-mers in output
         -b        print sequences and base qualities
         -l STR    log file name to output statistics
         -t INT    number of threads [1]
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-assignment">
<h2>10.4. <code class="docutils literal"><span class="pre">prophyle_assignment</span></code><a class="headerlink" href="#prophyle-assignment" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_assignment.py -h

usage: prophyle_assignment.py [-h] [-f {kraken,sam}] [-m {h1,c1,c2,h2}] [-A]
                              [-L] [-X] [-c [STR [STR ...]]]
                              &lt;tree.nhx&gt; &lt;k&gt; &lt;assignments.txt&gt;

Implementation of assignment algorithm

positional arguments:
  &lt;tree.nhx&gt;          phylogenetic tree (Newick/NHX)
  &lt;k&gt;                 k-mer length
  &lt;assignments.txt&gt;   assignments in generalized Kraken format

optional arguments:
  -h, --help          show this help message and exit
  -f {kraken,sam}     format of output [sam]
  -m {h1,c1,c2,h2}    measure: h1=hit count, c1=coverage, h2=norm.hit count,
                      c2=norm.coverage [h1]
  -A                  annotate assignments
  -L                  use LCA when tie (multiple assignments with the same
                      score)
  -X                  use LCA for k-mers (multiple hits of a k-mer)
  -c [STR [STR ...]]  configuration (a JSON dictionary)
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-analyze-py">
<h2>10.5. <code class="docutils literal"><span class="pre">prophyle_analyze.py</span></code><a class="headerlink" href="#prophyle-analyze-py" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_analyze.py -h

usage: prophyle_analyze.py [-h] [-s [&#39;w&#39;, &#39;u&#39;, &#39;wl&#39;, &#39;ul&#39;]]
                           [-f [&#39;sam&#39;, &#39;bam&#39;, &#39;cram&#39;, &#39;uncompressed_bam&#39;, &#39;kraken&#39;, &#39;histo&#39;]]
                           {index_dir, tree.nw} &lt;out_prefix&gt; &lt;input_fn&gt;
                           [&lt;input_fn&gt; ...]

Program: prophyle_analyze.py

Analyze results of ProPhyle&#39;s classification.
Stats:
w: weighted assignments
u: unique assignments (ignore multiple assignments)
wl: weighted assignments, propagated to leaves
ul: unique assignments, propagated to leaves

positional arguments:
  {index_dir, tree.nw}  Index directory or phylogenetic tree
  &lt;out_prefix&gt;          Prefix for output files (the complete file names will
                        be &lt;out_prefix&gt;_rawhits.tsv for the raw hit counts
                        table and &lt;out_prefix&gt;_otu.tsv for the otu table)
  &lt;input_fn&gt;            ProPhyle output files whose format is chosen with the
                        -f option. Use &#39;-&#39; for stdin or multiple files with
                        the same format (one per sample)

optional arguments:
  -h, --help            show this help message and exit
  -s [&#39;w&#39;, &#39;u&#39;, &#39;wl&#39;, &#39;ul&#39;]
                        Statistics to use for the computation of histograms: w
                        (default) =&gt; weighted assignments; u =&gt; unique
                        assignments, non-weighted; wl =&gt; weighted assignments,
                        propagated to leaves; ul =&gt; unique assignments,
                        propagated to leaves.
  -f [&#39;sam&#39;, &#39;bam&#39;, &#39;cram&#39;, &#39;uncompressed_bam&#39;, &#39;kraken&#39;, &#39;histo&#39;]
                        Input format of assignments [auto]. If &#39;histo&#39; is
                        selected the program expects hit count histograms
                        (*_rawhits.tsv) previously computed using prophyle
                        analyze, it merges them and compute OTU table from the
                        result (assignment files are not required)
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-propagation-makefile-py">
<h2>10.6. <code class="docutils literal"><span class="pre">prophyle_propagation_makefile.py</span></code><a class="headerlink" href="#prophyle-propagation-makefile-py" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_propagation_makefile.py -h

usage: prophyle_propagation_makefile.py [-h] -k int
                                        &lt;tree.nw&gt; &lt;library.dir&gt; &lt;output.dir&gt;
                                        &lt;Makefile&gt;

Create Makefile for parallelized ProPhyle k-mer propagation.

positional arguments:
  &lt;tree.nw&gt;      phylogenetic tree (in Newick/NHX).
  &lt;library.dir&gt;  directory with the library
  &lt;output.dir&gt;   output directory for the index
  &lt;Makefile&gt;     output Makefile

optional arguments:
  -h, --help     show this help message and exit
  -k int         k-mer length
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-propagation-preprocessing-py">
<h2>10.7. <code class="docutils literal"><span class="pre">prophyle_propagation_preprocessing.py</span></code><a class="headerlink" href="#prophyle-propagation-preprocessing-py" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_propagation_preprocessing.py -h

usage: prophyle_propagation_preprocessing.py [-h] [-s FLOAT] [-A] [-V] [-P]
                                             &lt;in_tree.nw{@node_name}&gt;
                                             [&lt;in_tree.nw{@node_name}&gt; ...]
                                             &lt;out_tree.nw&gt;

Merge multiple ProPhyle trees. Specific subtrees might be extracted before merging. Examples:
        $ prophyle_merge_trees.py ~/prophyle/bacteria.nw ~/prophyle/viruses.nw bv.nw
        $ prophyle_merge_trees.py ~/prophyle/bacteria.nw@562 ecoli.nw

positional arguments:
  &lt;in_tree.nw{@node_name}&gt;
                        input tree
  &lt;out_tree.nw&gt;         output tree

optional arguments:
  -h, --help            show this help message and exit
  -s FLOAT              rate of sampling the tree [no sampling]
  -A                    autocomplete tree (names of internal nodes and FASTA paths)
  -V                    verbose
  -P                    do not add prefixes to node names
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-propagation-postprocessing-py">
<h2>10.8. <code class="docutils literal"><span class="pre">prophyle_propagation_postprocessing.py</span></code><a class="headerlink" href="#prophyle-propagation-postprocessing-py" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_propagation_postprocessing.py -h

usage: prophyle_propagation_postprocessing.py [-h]
                                              &lt;propagation.dir&gt; &lt;index.fa&gt;
                                              &lt;in.tree.nw&gt; &lt;counts.tsv&gt;
                                              &lt;out.tree.nw&gt;

K-mer propagation postprocessing: merging FASTA files and k-mer annotation.

positional arguments:
  &lt;propagation.dir&gt;  directory with FASTA files
  &lt;index.fa&gt;         output fast file
  &lt;in.tree.nw&gt;       input phylogenetic tree
  &lt;counts.tsv&gt;       input phylogenetic tree
  &lt;out.tree.nw&gt;      output phylogenetic tree

optional arguments:
  -h, --help         show this help message and exit
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-validate-tree-py">
<h2>10.9. <code class="docutils literal"><span class="pre">prophyle_validate_tree.py</span></code><a class="headerlink" href="#prophyle-validate-tree-py" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_validate_tree.py -h

usage: prophyle_validate_tree.py [-h] &lt;tree.nw&gt; [&lt;tree.nw&gt; ...]

Verify a Newick/NHX tree

positional arguments:
  &lt;tree.nw&gt;   phylogenetic tree (in Newick/NHX)

optional arguments:
  -h, --help  show this help message and exit
</pre></div>
</div>
</div></blockquote>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="formats.html" title="9. File formats"
             >previous</a> |</li>
        <li><a href="index.html">ProPhyle home</a>&#160;|</li>
        <li><a href="contents.html">Documentation</a> &#187;</li>
 
      </ul>
    </div>


    <div class="footer" role="contentinfo">
        &#169; Copyright 2015-2018, Karel Břinda, Kamil Salikhov, Simone Pignotti, Gregory Kucherov.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.6.2.
    </div>

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-112241191-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-112241191-1');
</script>


  </body>
</html>