
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>5. Building a custom database &#8212; ProPhyle 0.3.2.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx13.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <link rel="search" type="application/opensearchdescription+xml"
          title="Search within ProPhyle 0.3.2.0 documentation"
          href="_static/opensearch.xml"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="6. Classifying reads" href="classify.html" />
    <link rel="prev" title="4. Building a standard database" href="standard_db.html" />
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:300,400,700'
          rel='stylesheet' type='text/css' />
 
    <style type="text/css">
      table.right { float: right; margin-left: 20px; }
      table.right td { border: 1px solid #ccc; }
      
    </style>
    <script type="text/javascript">
      // intelligent scrolling of the sidebar content
      $(window).scroll(function() {
        var sb = $('.sphinxsidebarwrapper');
        var win = $(window);
        var sbh = sb.height();
        var offset = $('.sphinxsidebar').position()['top'];
        var wintop = win.scrollTop();
        var winbot = wintop + win.innerHeight();
        var curtop = sb.position()['top'];
        var curbot = curtop + sbh;
        // does sidebar fit in window?
        if (sbh < win.innerHeight()) {
          // yes: easy case -- always keep at the top
          sb.css('top', $u.min([$u.max([0, wintop - offset - 10]),
                                $(document).height() - sbh - 200]));
        } else {
          // no: only scroll if top/bottom edge of sidebar is at
          // top/bottom edge of window
          if (curtop > wintop && curbot > winbot) {
            sb.css('top', $u.max([wintop - offset - 10, 0]));
          } else if (curtop < wintop && curbot < winbot) {
            sb.css('top', $u.min([winbot - sbh - offset - 20,
                                  $(document).height() - sbh - 200]));
          }
        }
      });
    </script>

  </head><body>
<div class="pageheader">
  <ul>
    <li><a href="index.html">Home</a></li>
    <li><a href="install.html">Get it</a></li>
    <li><a href="contents.html">Docs</a></li>
    <li><a href="http://github.com/prophyle/prophyle">Extend/Develop</a></li>
  </ul>
  <div>
    <a href="index.html">
      <!--
        <img src="_static/sphinxheader.png" alt="SPHINX" />
      -->
      <div style="color:white;font-size:240%">ProPhyle</div>
      <div style="color:white;font-size:140%">DNA sequence classification</div>
    </a>
  </div>
</div>

    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="classify.html" title="6. Classifying reads"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="standard_db.html" title="4. Building a standard database"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">ProPhyle home</a>&#160;|</li>
        <li><a href="contents.html">Documentation</a> &#187;</li>

        <li class="nav-item nav-item-this"><a href=""><span class="section-number">5. </span>Building a custom database</a></li> 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="contents.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">5. Building a custom database</a><ul>
<li><a class="reference internal" href="#building-a-custom-database-with-user-provided-data">5.1. Building a custom database with user-provided data</a></li>
<li><a class="reference internal" href="#building-a-custom-database-with-refseq-genomes-and-the-ncbi-taxonomy">5.2. Building a custom database with RefSeq genomes and the NCBI taxonomy</a><ul>
<li><a class="reference internal" href="#download-sequences-from-ncbi">Download sequences from NCBI</a></li>
<li><a class="reference internal" href="#build-a-tree">Build a tree</a></li>
<li><a class="reference internal" href="#build-the-index">Build the index</a></li>
</ul>
</li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="standard_db.html"
                        title="previous chapter"><span class="section-number">4. </span>Building a standard database</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="classify.html"
                        title="next chapter"><span class="section-number">6. </span>Classifying reads</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/custom_db.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="building-a-custom-database">
<span id="custom-db"></span><h1><span class="section-number">5. </span>Building a custom database<a class="headerlink" href="#building-a-custom-database" title="Permalink to this headline">¶</a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#building-a-custom-database-with-user-provided-data" id="id3">Building a custom database with user-provided data</a></p></li>
<li><p><a class="reference internal" href="#building-a-custom-database-with-refseq-genomes-and-the-ncbi-taxonomy" id="id4">Building a custom database with RefSeq genomes and the NCBI taxonomy</a></p>
<ul>
<li><p><a class="reference internal" href="#download-sequences-from-ncbi" id="id5">Download sequences from NCBI</a></p></li>
<li><p><a class="reference internal" href="#build-a-tree" id="id6">Build a tree</a></p></li>
<li><p><a class="reference internal" href="#build-the-index" id="id7">Build the index</a></p></li>
</ul>
</li>
</ul>
</div>
<section id="building-a-custom-database-with-user-provided-data">
<h2><span class="section-number">5.1. </span>Building a custom database with user-provided data<a class="headerlink" href="#building-a-custom-database-with-user-provided-data" title="Permalink to this headline">¶</a></h2>
</section>
<section id="building-a-custom-database-with-refseq-genomes-and-the-ncbi-taxonomy">
<h2><span class="section-number">5.2. </span>Building a custom database with RefSeq genomes and the NCBI taxonomy<a class="headerlink" href="#building-a-custom-database-with-refseq-genomes-and-the-ncbi-taxonomy" title="Permalink to this headline">¶</a></h2>
<section id="download-sequences-from-ncbi">
<h3>Download sequences from NCBI<a class="headerlink" href="#download-sequences-from-ncbi" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>Create a subdirectory in ProPhyle’s main directory, and change the working directory to the newly created one</p></li>
<li><p>Download an <code class="docutils literal notranslate"><span class="pre">assembly_summary.txt</span></code> file from NCBI’s FTP server. For RefSeq’s bacterial genomes, follow this <a class="reference external" href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt">link</a>. More information about these files can be found <a class="reference external" href="ftp://ftp.ncbi.nlm.nih.gov/genomes/README_assembly_summary.txt">here</a></p></li>
<li><p>Select genomes of interest; the following command will select all complete genomes, but it is sufficient to add more conditions to <code class="docutils literal notranslate"><span class="pre">awk</span></code> to select a subset. Fields 1, 6 and 20 contain respectively the accession number, taxonomic identifier and ftp directory of each genome. <code class="docutils literal notranslate"><span class="pre">acc2taxid.tsv</span></code> will therefore be a tab separated file containing the taxid corresponding to each sequence’s accession number. Please refer to <a class="reference external" href="https://www.ncbi.nlm.nih.gov/genome/doc/ftpfaq">NCBI’s genomes download FAQ</a> for further information)</p></li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>awk -F <span class="s2">&quot;\t&quot;</span> -v <span class="nv">OFS</span><span class="o">=</span><span class="s2">&quot;\t&quot;</span> <span class="s1">&#39;$12==&quot;Complete Genome&quot; &amp;&amp; $11==&quot;latest&quot;\</span>
<span class="s1">        {print $1, $6, $20}&#39;</span> assembly_summary.txt &gt;ftpselection.tsv
cut -f <span class="m">3</span> ftpselection.tsv <span class="p">|</span> awk <span class="s1">&#39;BEGIN{FS=OFS=&quot;/&quot;;filesuffix=&quot;genomic.fna.gz&quot;}\</span>
<span class="s1">        {ftpdir=$0;asm=$10;file=asm&quot;_&quot;filesuffix;print ftpdir,file}&#39;</span> &gt;ftpfilepaths.tsv
cut -f <span class="m">1</span>,2 ftpselection.tsv &gt;acc2taxid.tsv
</pre></div>
</div>
<ul class="simple">
<li><p>Download selected sequences using <code class="docutils literal notranslate"><span class="pre">parallel</span> <span class="pre">--gnu</span> <span class="pre">-j</span> <span class="pre">24</span> <span class="pre">-a</span> <span class="pre">ftpfilepaths.tsv</span> <span class="pre">wget</span></code>. Number of jobs (<code class="docutils literal notranslate"><span class="pre">-j</span></code> option) should be changed according to the number of      available cores and bandwidth.</p></li>
<li><p>Extract them using <code class="docutils literal notranslate"><span class="pre">parallel</span> <span class="pre">--gnu</span> <span class="pre">-j</span> <span class="pre">24</span> <span class="pre">-a</span> <span class="pre">ftpfilepaths.tsv</span> <span class="pre">gunzip</span> <span class="pre">{/}</span></code></p></li>
<li><p>Create a <a class="reference external" href="http://www.htslib.org/doc/faidx.html">fasta index</a> for each file using <code class="docutils literal notranslate"><span class="pre">find</span> <span class="pre">.</span> <span class="pre">-name</span> <span class="pre">'*.fna'</span> <span class="pre">|</span> <span class="pre">parallel</span> <span class="pre">--no-notice</span> <span class="pre">--verbose</span> <span class="pre">samtools</span> <span class="pre">faidx</span> <span class="pre">{}</span></code></p></li>
</ul>
</section>
<section id="build-a-tree">
<h3>Build a tree<a class="headerlink" href="#build-a-tree" title="Permalink to this headline">¶</a></h3>
<p>Build a taxonomic tree for the downloaded sequences using</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>prophyle_homedir/prophyle/prophyle_ncbitree.py &lt;library_name&gt; &lt;library_main_dir&gt;<span class="se">\</span>
        &lt;output_file&gt; acc2taxid.tsv -l &lt;log_file&gt;
</pre></div>
</div>
<p>Taxonomic identifiers are assigned to the sequences first, and then the tree is
built using <a class="reference external" href="http://etetoolkit.org/">ETE Toolkit</a> and saved with newick format
1. Necessary node attributes are:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">name</span></code>: unique node name (ideally the TaxID of the node)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">taxid</span></code>: unique taxonomic identifier</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">seqname</span></code>: names of the sequences sharing the same taxid, separated by &#64;</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">fastapath</span></code>: paths of the sequences’ fasta files, separated by &#64; (relative paths from ProPhyle’s home directory)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">infasta_offset</span></code>: positions where each sequence starts inside the corresponding fasta files, separated by &#64;</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">base_len</span></code>: length of each sequence, separated by &#64;</p></li>
</ul>
<p><code class="docutils literal notranslate"><span class="pre">seqname</span></code>, <code class="docutils literal notranslate"><span class="pre">infasta_offset</span></code> and <code class="docutils literal notranslate"><span class="pre">base_len</span></code> can be found in samtools’
<a class="reference external" href="http://www.htslib.org/doc/faidx.html">fasta index</a>. Other optional attributes
are <code class="docutils literal notranslate"><span class="pre">sci_name</span></code>, <code class="docutils literal notranslate"><span class="pre">named_lineage</span></code>, <code class="docutils literal notranslate"><span class="pre">lineage</span></code>, <code class="docutils literal notranslate"><span class="pre">rank</span></code> (more info
<a class="reference external" href="http://etetoolkit.org/docs/latest/tutorial/tutorial_ncbitaxonomy.html#automatic-tree-annotation-using-ncbi-taxonomy">here</a>
). Using ETE library or modifying the <code class="docutils literal notranslate"><span class="pre">prophyle_ncbitree.py</span></code> script it is
easy to adapt any tree to match the requirements above.</p>
<p>If taxonomic information is not available, <code class="docutils literal notranslate"><span class="pre">taxid</span></code> attribute can be replaced by the node name. This could be useful to perform classification over custom phylogenetic trees.</p>
</section>
<section id="build-the-index">
<h3>Build the index<a class="headerlink" href="#build-the-index" title="Permalink to this headline">¶</a></h3>
<p>Run the standard command to build ProPhyle index</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>prophyle index -k &lt;kmer_length&gt; &lt;tree_1.nw&gt; <span class="o">[</span>&lt;tree_2.nw&gt; ...<span class="o">]</span> &lt;index_dir&gt;
</pre></div>
</div>
</section>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="classify.html" title="6. Classifying reads"
             >next</a> |</li>
        <li class="right" >
          <a href="standard_db.html" title="4. Building a standard database"
             >previous</a> |</li>
        <li><a href="index.html">ProPhyle home</a>&#160;|</li>
        <li><a href="contents.html">Documentation</a> &#187;</li>

        <li class="nav-item nav-item-this"><a href=""><span class="section-number">5. </span>Building a custom database</a></li> 
      </ul>
    </div>


    <div class="footer" role="contentinfo">
        &#169; Copyright 2015-2021, Karel Břinda, Kamil Salikhov, Simone Pignotti, Gregory Kucherov.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 4.1.2.
    </div>

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-112241191-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-112241191-1');
</script>


  </body>
</html>