
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  
  <!-- Licensed under the Apache 2.0 License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/open-sans/stylesheet.css" />
  <!-- Licensed under the SIL Open Font License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/source-serif-pro/source-serif-pro.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap-theme.min.css" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
    <title>File formats &#8212; ProPhyle  documentation</title>
    <link rel="stylesheet" href="_static/guzzle.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href=".static/prophyle.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Command-line interface" href="reference.html" />
    <link rel="prev" title="Read classification" href="classification.html" />
  
   

  </head>
  <body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="reference.html" title="Command-line interface"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="classification.html" title="Read classification"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">ProPhyle  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="container-wrapper">

      <div id="mobile-toggle">
        <a href="#"><span class="glyphicon glyphicon-align-justify" aria-hidden="true"></span></a>
      </div>
  <div id="left-column">
    <div class="sphinxsidebar">
        <a href="
    index.html" class="text-logo">ProPhyle</a>
        
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <h2>Table Of Contents</h2>
  </div>
  <div class="sidebar-toc">
    
    
      <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="index.html">ProPhyle</a></li>
<li class="toctree-l1"><a class="reference internal" href="example.html">Quick example</a></li>
<li class="toctree-l1"><a class="reference internal" href="requirements.html">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="install.html">Installation and running</a></li>
<li class="toctree-l1"><a class="reference internal" href="databases_predefined.html">Building predefined databases</a></li>
<li class="toctree-l1"><a class="reference internal" href="databases_custom.html">Building custom databases</a></li>
<li class="toctree-l1"><a class="reference internal" href="classification.html">Read classification</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">File formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="reference.html">Command-line interface</a></li>
</ul>

    
  </div>
</div>
        
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <div id="main-search">
      <form class="form-inline" action="search.html" method="GET" role="form">
        <div class="input-group">
          <input name="q" type="text" class="form-control" placeholder="Search...">
        </div>
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
      </form>
    </div>
  </div>
</div>
    </div>
  </div>
        <div id="right-column">
          
          <div role="navigation" aria-label="breadcrumbs navigation">
            <ol class="breadcrumb">
              <li><a href="index.html">Docs</a></li>
              
              <li>File formats</li>
            </ol>
          </div>
          
          <div class="document clearer body">
            
  <div class="section" id="file-formats">
<span id="formats"></span><h1>File formats<a class="headerlink" href="#file-formats" title="Permalink to this headline">¶</a></h1>
<div class="section" id="input-trees">
<h2>Input trees<a class="headerlink" href="#input-trees" title="Permalink to this headline">¶</a></h2>
<div class="section" id="newick">
<h3>Newick<a class="headerlink" href="#newick" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="http://evolution.genetics.washington.edu/phylip/newicktree.html">Newick</a> trees are eventually transformed to NHX.</p>
</div>
<div class="section" id="new-hampshire-x-format-nhx">
<h3>New Hampshire X Format (NHX)<a class="headerlink" href="#new-hampshire-x-format-nhx" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://sites.google.com/site/cmzmasek/home/software/forester/nhx">NHX format</a>
is parsed using the <a class="reference external" href="http://etetoolkit.org/">ETE3 library</a>  (see specification of <a class="reference external" href="http://etetoolkit.org/docs/latest/tutorial/tutorial_trees.html#reading-and-writing-newick-trees">format 1</a>).</p>
</div>
</div>
<div class="section" id="classification-output">
<h2>Classification output<a class="headerlink" href="#classification-output" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="http://samtools.github.io/hts-specs/">SAM format</a></p>
<div class="section" id="kraken-like-format">
<h3>Kraken-like format<a class="headerlink" href="#kraken-like-format" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://ccb.jhu.edu/software/kraken/MANUAL.html#output-format">Kraken</a></p>
</div>
</div>
<div class="section" id="abundences">
<h2>Abundences<a class="headerlink" href="#abundences" title="Permalink to this headline">¶</a></h2>
<div class="section" id="kraken-report-format">
<h3>Kraken-report format<a class="headerlink" href="#kraken-report-format" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://ccb.jhu.edu/software/kraken/MANUAL.html#sample-reports">kraken-report</a> format:</p>
<blockquote>
<div><ol class="arabic simple">
<li>Percentage of reads covered by the clade rooted at this taxon</li>
<li>Number of reads covered by the clade rooted at this taxon</li>
<li>Number of reads assigned directly to this taxon</li>
<li>A rank code, indicating (U)nclassified, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. All other ranks are simply ‘-‘.</li>
<li>NCBI taxonomy ID</li>
<li>indented scientific name</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="metaphlan2-format">
<h3>Metaphlan2 format<a class="headerlink" href="#metaphlan2-format" title="Permalink to this headline">¶</a></h3>
<dl class="docutils">
<dt><a class="reference external" href="https://bitbucket.org/biobakery/biobakery/wiki/metaphlan2#rst-header-output-files">MetaPhlAn2</a> format:</dt>
<dd><ol class="first arabic simple">
<li>clades, ranging from taxonomic kingdoms (Bacteria, Archaea, etc.) through species.</li>
</ol>
<p class="last">2. The taxonomic level of each clade is prefixed to indicate its level: Kingdom: <code class="docutils literal"><span class="pre">k__</span></code>, Phylum: <code class="docutils literal"><span class="pre">p__</span></code>, Class: <code class="docutils literal"><span class="pre">c__</span></code>, Order: <code class="docutils literal"><span class="pre">o__</span></code>, Family: <code class="docutils literal"><span class="pre">f__</span></code>, Genus: <code class="docutils literal"><span class="pre">g__</span></code>, Species: <code class="docutils literal"><span class="pre">s__</span></code>.
Since sequence-based profiling is relative and does not provide absolute cellular abundance measures,
clades are hierarchically summed. Each level will sum to 100%; that is, the sum of all kindom-level
clades is 100%, the sum of all genus-level clades (including unclassified) is also 100%, and so forth.
OTU equivalents can be extracted by using only the species-level <code class="docutils literal"><span class="pre">s__</span></code> clades from this file
(again, making sure to include clades unclassified at this level).</p>
</dd>
</dl>
</div>
<div class="section" id="centrifuge-format">
<h3>Centrifuge format<a class="headerlink" href="#centrifuge-format" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://ccb.jhu.edu/software/centrifuge/manual.shtml#centrifuge-summary-output-the-default-filename-is-centrifuge_report.tsv">Centrifuge</a> format:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="c1">#name                                                           taxID   taxRank    kmerCount   numReads   numUniqueReads   abundance</span>
<span class="n">Wigglesworthia</span> <span class="n">glossinidia</span> <span class="n">endosymbiont</span> <span class="n">of</span> <span class="n">Glossina</span> <span class="n">brevipalpis</span> <span class="mi">36870</span>   <span class="n">leaf</span>       <span class="mi">703004</span>      <span class="mf">5981.37</span>    <span class="mi">5964</span>             <span class="mi">0</span>
</pre></div>
</div>
<ol class="arabic simple">
<li>name of a genome, or the name corresponding to a taxonomic ID (the second column) at a rank higher than the strain (e.g., Wigglesworthia glossinidia endosymbiont of Glossina brevipalpis).</li>
<li>taxonomic ID (e.g., 36870).</li>
<li>taxonomic rank (e.g., leaf).</li>
<li>number of k-mers propagated up to the node (e.g., 703004).</li>
<li>number of reads classified to this node including multi-classified reads (divided by the number of assignments, e.g., 5981.37).</li>
<li>number of reads uniquely classified to this genomic sequence (e.g., 5964).</li>
<li>not used yet.</li>
</ol>
</div>
</div>
<div class="section" id="index">
<h2>Index<a class="headerlink" href="#index" title="Permalink to this headline">¶</a></h2>
<div class="section" id="standard">
<h3>Standard<a class="headerlink" href="#standard" title="Permalink to this headline">¶</a></h3>
<p>Directory with the following files:</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">index.fa</span></code> – assembled contigs, name of sequences are of the following format: <code class="docutils literal"><span class="pre">[node_name]&#64;c[contig_number]</span></code></li>
<li><code class="docutils literal"><span class="pre">index.fa.amb</span></code> – list of ambiguous nucleotides, no values</li>
<li><code class="docutils literal"><span class="pre">index.fa.ann</span></code> – list of contigs and their starting positions in the master string</li>
<li><code class="docutils literal"><span class="pre">index.fa.[k].klcp</span></code> – k-LCP array</li>
<li><code class="docutils literal"><span class="pre">index.fa.bwt</span></code> – Burrows-Wheeler Transform of the master string (merged sequences + reverse completement) + OCC table (BWA format)</li>
<li><code class="docutils literal"><span class="pre">index.fa.kmers.tsv</span></code> – k-mer statistics, format: <code class="docutils literal"><span class="pre">[node_name].[full|reduced].fa</span>&#160;&#160;&#160; <span class="pre">[#kmers]</span></code>, where <code class="docutils literal"><span class="pre">full</span></code> refers to all associated k-mers and <code class="docutils literal"><span class="pre">reduced</span></code> to represented k-mers</li>
<li><code class="docutils literal"><span class="pre">index.fa.pac</span></code> – packed sequences (BWA format)</li>
<li><code class="docutils literal"><span class="pre">index.fa.sa</span></code> – sampled suffix array (BWA format)</li>
<li><code class="docutils literal"><span class="pre">index.json</span></code> – JSON dictionary with a k-mer size (<code class="docutils literal"><span class="pre">k</span></code>) and ProPhyle version specification (<code class="docutils literal"><span class="pre">prophyle-version</span></code>, <code class="docutils literal"><span class="pre">prophyle-revision</span></code>, <code class="docutils literal"><span class="pre">prophyle-commit</span></code>)</li>
<li><code class="docutils literal"><span class="pre">log.txt</span></code></li>
<li><code class="docutils literal"><span class="pre">tree.nw</span></code> – phylogenetic tree adjusted for classification</li>
<li><code class="docutils literal"><span class="pre">tree.preliminary.nw</span></code> – phylogenetic tree before adjusting</li>
</ul>
</div>
<div class="section" id="compressed-for-transmission-only">
<h3>Compressed (for transmission only)<a class="headerlink" href="#compressed-for-transmission-only" title="Permalink to this headline">¶</a></h3>
<p>Archive <code class="docutils literal"><span class="pre">[name].tar.gz</span></code> with the following subset of index files:</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">index.fa.amb</span></code></li>
<li><code class="docutils literal"><span class="pre">index.fa.ann</span></code></li>
<li><code class="docutils literal"><span class="pre">index.fa.bwt</span></code> – Burrows-Wheeler Transform <em>without</em> the OCC table</li>
<li><code class="docutils literal"><span class="pre">index.fa.kmers.tsv</span></code></li>
<li><code class="docutils literal"><span class="pre">index.json</span></code></li>
<li><code class="docutils literal"><span class="pre">tree.nw</span></code></li>
<li><code class="docutils literal"><span class="pre">tree.preliminary.nw</span></code></li>
</ul>
</div>
</div>
</div>


          </div>
            
  <div class="footer-relations">
    
      <div class="pull-left">
        <a class="btn btn-default" href="classification.html" title="previous chapter (use the left arrow)">Read classification</a>
      </div>
    
      <div class="pull-right">
        <a class="btn btn-default" href="reference.html" title="next chapter (use the right arrow)">Command-line interface</a>
      </div>
    </div>
    <div class="clearer"></div>
  
        </div>
        <div class="clearfix"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="reference.html" title="Command-line interface"
             >next</a> |</li>
        <li class="right" >
          <a href="classification.html" title="Read classification"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">ProPhyle  documentation</a> &#187;</li> 
      </ul>
    </div>
<script type="text/javascript">
  $("#mobile-toggle a").click(function () {
    $("#left-column").toggle();
  });
</script>
<script type="text/javascript" src="_static/js/bootstrap.js"></script>
  <div class="footer">
    &copy; Copyright 2016–2017, Karel Břinda, Kamil Salikhov, Simone Pignotti, Gregory Kucherov. Created using <a href="http://sphinx.pocoo.org/">Sphinx</a>.
  </div>
  </body>
</html>