
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>File formats &#8212; ProPhyle 0.3.1.0 documentation</title>
    <link rel="stylesheet" href="_static/sphinx13.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.3.1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="search" type="application/opensearchdescription+xml"
          title="Search within ProPhyle 0.3.1.0 documentation"
          href="_static/opensearch.xml"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Command-line interface" href="cli.html" />
    <link rel="prev" title="Compressing indexes for transmission" href="compress.html" />
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:300,400,700'
          rel='stylesheet' type='text/css' />
 
    <style type="text/css">
      table.right { float: right; margin-left: 20px; }
      table.right td { border: 1px solid #ccc; }
      
    </style>
    <script type="text/javascript">
      // intelligent scrolling of the sidebar content
      $(window).scroll(function() {
        var sb = $('.sphinxsidebarwrapper');
        var win = $(window);
        var sbh = sb.height();
        var offset = $('.sphinxsidebar').position()['top'];
        var wintop = win.scrollTop();
        var winbot = wintop + win.innerHeight();
        var curtop = sb.position()['top'];
        var curbot = curtop + sbh;
        // does sidebar fit in window?
        if (sbh < win.innerHeight()) {
          // yes: easy case -- always keep at the top
          sb.css('top', $u.min([$u.max([0, wintop - offset - 10]),
                                $(document).height() - sbh - 200]));
        } else {
          // no: only scroll if top/bottom edge of sidebar is at
          // top/bottom edge of window
          if (curtop > wintop && curbot > winbot) {
            sb.css('top', $u.max([wintop - offset - 10, 0]));
          } else if (curtop < wintop && curbot < winbot) {
            sb.css('top', $u.min([winbot - sbh - offset - 20,
                                  $(document).height() - sbh - 200]));
          }
        }
      });
    </script>

  </head>
  <body>
<div class="pageheader">
  <ul>
    <li><a href="index.html">Home</a></li>
    <li><a href="install.html">Get it</a></li>
    <li><a href="contents.html">Docs</a></li>
    <li><a href="http://github.com/prophyle/prophyle">Extend/Develop</a></li>
  </ul>
  <div>
    <a href="index.html">
      <!--
        <img src="_static/sphinxheader.png" alt="SPHINX" />
      -->
      <div style="color:white;font-size:240%">ProPhyle</div>
      <div style="color:white;font-size:140%">DNA sequence classification</div>
    </a>
  </div>
</div>

    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="cli.html" title="Command-line interface"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="compress.html" title="Compressing indexes for transmission"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">ProPhyle home</a>&#160;|</li>
        <li><a href="contents.html">Documentation</a> &#187;</li>
 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="contents.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">File formats</a><ul>
<li><a class="reference internal" href="#input-data-formats">Input data formats</a><ul>
<li><a class="reference internal" href="#newick-trees">Newick trees</a><ul>
<li><a class="reference internal" href="#examples">Examples</a></li>
</ul>
</li>
<li><a class="reference internal" href="#nhx-trees">NHX trees</a><ul>
<li><a class="reference internal" href="#example">Example</a></li>
</ul>
</li>
<li><a class="reference internal" href="#sequences">Sequences</a></li>
</ul>
</li>
<li><a class="reference internal" href="#assignments">Assignments</a><ul>
<li><a class="reference internal" href="#read-assignments-in-sam-bam">Read assignments in SAM/BAM</a></li>
<li><a class="reference internal" href="#read-assignments-in-a-kraken-like-format">Read assignments in a Kraken-like format</a></li>
</ul>
</li>
<li><a class="reference internal" href="#abundances-estimates-experimental">Abundances estimates (experimental)</a><ul>
<li><a class="reference internal" href="#abundances-in-the-kraken-report-format">Abundances in the Kraken report format</a></li>
<li><a class="reference internal" href="#abundances-in-the-metaphlan2-report-format">Abundances in the Metaphlan2 report format</a></li>
<li><a class="reference internal" href="#abundances-in-the-centrifuge-report-format">Abundances in the Centrifuge report format</a><ul>
<li><a class="reference internal" href="#specification">Specification</a></li>
<li><a class="reference internal" href="#id1">Example</a></li>
</ul>
</li>
</ul>
</li>
<li><a class="reference internal" href="#internal-prophyle-formats">Internal ProPhyle formats</a><ul>
<li><a class="reference internal" href="#prophyle-index">ProPhyle Index</a></li>
<li><a class="reference internal" href="#compressed-prophyle-index-for-transmission">Compressed ProPhyle index for transmission</a></li>
</ul>
</li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="compress.html"
                        title="previous chapter">Compressing indexes for transmission</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="cli.html"
                        title="next chapter">Command-line interface</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/formats.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="file-formats">
<span id="formats"></span><h1>File formats<a class="headerlink" href="#file-formats" title="Permalink to this headline">¶</a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><a class="reference internal" href="#input-data-formats" id="id10">Input data formats</a><ul>
<li><a class="reference internal" href="#newick-trees" id="id11">Newick trees</a><ul>
<li><a class="reference internal" href="#examples" id="id12">Examples</a></li>
</ul>
</li>
<li><a class="reference internal" href="#nhx-trees" id="id13">NHX trees</a><ul>
<li><a class="reference internal" href="#example" id="id14">Example</a></li>
</ul>
</li>
<li><a class="reference internal" href="#sequences" id="id15">Sequences</a></li>
</ul>
</li>
<li><a class="reference internal" href="#assignments" id="id16">Assignments</a><ul>
<li><a class="reference internal" href="#read-assignments-in-sam-bam" id="id17">Read assignments in SAM/BAM</a></li>
<li><a class="reference internal" href="#read-assignments-in-a-kraken-like-format" id="id18">Read assignments in a Kraken-like format</a></li>
</ul>
</li>
<li><a class="reference internal" href="#abundances-estimates-experimental" id="id19">Abundances estimates (experimental)</a><ul>
<li><a class="reference internal" href="#abundances-in-the-kraken-report-format" id="id20">Abundances in the Kraken report format</a></li>
<li><a class="reference internal" href="#abundances-in-the-metaphlan2-report-format" id="id21">Abundances in the Metaphlan2 report format</a></li>
<li><a class="reference internal" href="#abundances-in-the-centrifuge-report-format" id="id22">Abundances in the Centrifuge report format</a><ul>
<li><a class="reference internal" href="#specification" id="id23">Specification</a></li>
<li><a class="reference internal" href="#id1" id="id24">Example</a></li>
</ul>
</li>
</ul>
</li>
<li><a class="reference internal" href="#internal-prophyle-formats" id="id25">Internal ProPhyle formats</a><ul>
<li><a class="reference internal" href="#prophyle-index" id="id26">ProPhyle Index</a></li>
<li><a class="reference internal" href="#compressed-prophyle-index-for-transmission" id="id27">Compressed ProPhyle index for transmission</a></li>
</ul>
</li>
</ul>
</div>
<div class="section" id="input-data-formats">
<h2>Input data formats<a class="headerlink" href="#input-data-formats" title="Permalink to this headline">¶</a></h2>
<div class="section" id="newick-trees">
<h3>Newick trees<a class="headerlink" href="#newick-trees" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="http://evolution.genetics.washington.edu/phylip/newicktree.html">Newick</a> trees are eventually transformed to NHX.</p>
<div class="section" id="examples">
<h4>Examples<a class="headerlink" href="#examples" title="Permalink to this headline">¶</a></h4>
<blockquote>
<div><p>A Newick tree with named leaves:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="p">((</span><span class="n">n1</span><span class="p">,</span><span class="n">n2</span><span class="p">,</span><span class="n">n3</span><span class="p">),(</span><span class="n">n5</span><span class="p">,</span><span class="n">n6</span><span class="p">));</span>
</pre></div>
</div>
<p>A Newick tree with nameds nodes:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="p">((</span><span class="n">n1</span><span class="p">,</span><span class="n">n2</span><span class="p">)</span><span class="n">o1</span><span class="p">,(</span><span class="n">n3</span><span class="p">,</span><span class="n">n4</span><span class="p">,</span><span class="n">n5</span><span class="p">)</span><span class="n">o2</span><span class="p">)</span><span class="n">p1</span><span class="p">;</span>
</pre></div>
</div>
</div></blockquote>
</div>
</div>
<div class="section" id="nhx-trees">
<h3>NHX trees<a class="headerlink" href="#nhx-trees" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://sites.google.com/site/cmzmasek/home/software/forester/nhx">New Hampshire X Format</a>
is parsed using the <a class="reference external" href="http://etetoolkit.org/">ETE3 library</a>  (see specification of <a class="reference external" href="http://etetoolkit.org/docs/latest/tutorial/tutorial_trees.html#reading-and-writing-newick-trees">format 1</a>).</p>
<blockquote>
<div><table border="1" class="colwidths-given docutils" id="id2">
<caption><span class="caption-text">NHX attributes</span><a class="headerlink" href="#id2" title="Permalink to this table">¶</a></caption>
<colgroup>
<col width="21%" />
<col width="21%" />
<col width="59%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Attribute</th>
<th class="head">Type</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>name</td>
<td>string</td>
<td>unique node name (ideally the TaxID of the node)</td>
</tr>
<tr class="row-odd"><td>taxid</td>
<td>&#160;</td>
<td>unique taxonomic identifier</td>
</tr>
<tr class="row-even"><td>seqname</td>
<td>&#160;</td>
<td>names of the sequences sharing the same taxid, separated by &#64;</td>
</tr>
<tr class="row-odd"><td>fastapath</td>
<td>&#160;</td>
<td>paths of the sequences’ fasta files, separated by &#64; (relative paths from ProPhyle’s home directory)</td>
</tr>
<tr class="row-even"><td>base_len</td>
<td>&#160;</td>
<td>length of each sequence, separated by &#64;</td>
</tr>
</tbody>
</table>
</div></blockquote>
<div class="section" id="example">
<h4>Example<a class="headerlink" href="#example" title="Permalink to this headline">¶</a></h4>
<blockquote>
<div><p>Previous tree after autocomplete:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="p">(((</span><span class="n">n1</span><span class="p">:</span><span class="mi">1</span><span class="p">[</span><span class="o">&amp;&amp;</span><span class="n">NHX</span><span class="p">:</span><span class="n">dist</span><span class="o">=</span><span class="mf">1.0</span><span class="p">:</span><span class="n">fastapath</span><span class="o">=</span><span class="n">n1</span><span class="o">.</span><span class="n">fa</span><span class="p">:</span><span class="n">support</span><span class="o">=</span><span class="mf">1.0</span><span class="p">],</span><span class="n">n2</span><span class="p">:</span><span class="mi">1</span><span class="p">[</span><span class="o">&amp;&amp;</span><span class="n">NHX</span><span class="p">:</span><span class="n">dist</span><span class="o">=</span><span class="mf">1.0</span><span class="p">:</span><span class="n">fastapath</span><span class="o">=</span><span class="n">n2</span><span class="o">.</span><span class="n">fa</span><span class="p">:</span><span class="n">support</span><span class="o">=</span><span class="mf">1.0</span><span class="p">])</span><span class="n">o1</span><span class="p">:</span><span class="mi">1</span><span class="p">[</span><span class="o">&amp;&amp;</span><span class="n">NHX</span><span class="p">:</span><span class="n">dist</span><span class="o">=</span><span class="mf">1.0</span><span class="p">:</span><span class="n">support</span><span class="o">=</span><span class="mf">1.0</span><span class="p">],(</span><span class="n">n3</span><span class="p">:</span><span class="mi">1</span><span class="p">[</span><span class="o">&amp;&amp;</span><span class="n">NHX</span><span class="p">:</span><span class="n">dist</span><span class="o">=</span><span class="mf">1.0</span><span class="p">:</span><span class="n">fastapath</span><span class="o">=</span><span class="n">n3</span><span class="o">.</span><span class="n">fa</span><span class="p">:</span><span class="n">support</span><span class="o">=</span><span class="mf">1.0</span><span class="p">],</span><span class="n">n4</span><span class="p">:</span><span class="mi">1</span><span class="p">[</span><span class="o">&amp;&amp;</span><span class="n">NHX</span><span class="p">:</span><span class="n">dist</span><span class="o">=</span><span class="mf">1.0</span><span class="p">:</span><span class="n">fastapath</span><span class="o">=</span><span class="n">n4</span><span class="o">.</span><span class="n">fa</span><span class="p">:</span><span class="n">support</span><span class="o">=</span><span class="mf">1.0</span><span class="p">],</span><span class="n">n5</span><span class="p">:</span><span class="mi">1</span><span class="p">[</span><span class="o">&amp;&amp;</span><span class="n">NHX</span><span class="p">:</span><span class="n">dist</span><span class="o">=</span><span class="mf">1.0</span><span class="p">:</span><span class="n">fastapath</span><span class="o">=</span><span class="n">n5</span><span class="o">.</span><span class="n">fa</span><span class="p">:</span><span class="n">support</span><span class="o">=</span><span class="mf">1.0</span><span class="p">])</span><span class="n">o2</span><span class="p">:</span><span class="mi">1</span><span class="p">[</span><span class="o">&amp;&amp;</span><span class="n">NHX</span><span class="p">:</span><span class="n">dist</span><span class="o">=</span><span class="mf">1.0</span><span class="p">:</span><span class="n">support</span><span class="o">=</span><span class="mf">1.0</span><span class="p">])</span><span class="n">p1</span><span class="p">:</span><span class="mi">0</span><span class="p">[</span><span class="o">&amp;&amp;</span><span class="n">NHX</span><span class="p">:</span><span class="n">dist</span><span class="o">=</span><span class="mf">0.0</span><span class="p">:</span><span class="n">support</span><span class="o">=</span><span class="mf">1.0</span><span class="p">])</span><span class="n">merge_root</span><span class="p">:</span><span class="mi">1</span><span class="p">[</span><span class="o">&amp;&amp;</span><span class="n">NHX</span><span class="p">:</span><span class="n">dist</span><span class="o">=</span><span class="mf">1.0</span><span class="p">:</span><span class="n">support</span><span class="o">=</span><span class="mf">1.0</span><span class="p">];</span>
</pre></div>
</div>
</div></blockquote>
</div>
</div>
<div class="section" id="sequences">
<h3>Sequences<a class="headerlink" href="#sequences" title="Permalink to this headline">¶</a></h3>
<p>Input sequences can be provided in the FASTA or FASTQ formats. Any non-ACGT characters are treated as
unknown nucleotides. All k-mers containing an unknown nucleotide are discarded.</p>
</div>
</div>
<div class="section" id="assignments">
<h2>Assignments<a class="headerlink" href="#assignments" title="Permalink to this headline">¶</a></h2>
<div class="section" id="read-assignments-in-sam-bam">
<h3>Read assignments in SAM/BAM<a class="headerlink" href="#read-assignments-in-sam-bam" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="http://samtools.github.io/hts-specs/">SAM format</a></p>
<blockquote>
<div><table border="1" class="colwidths-given docutils" id="id3">
<caption><span class="caption-text">SAM fields</span><a class="headerlink" href="#id3" title="Permalink to this table">¶</a></caption>
<colgroup>
<col width="12%" />
<col width="12%" />
<col width="77%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Column</th>
<th class="head">Name</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>1</td>
<td>QNAME</td>
<td>Query name</td>
</tr>
<tr class="row-odd"><td>2</td>
<td>FLAG</td>
<td><code class="docutils literal"><span class="pre">0</span></code> if assigned, <code class="docutils literal"><span class="pre">4</span></code> otherwise</td>
</tr>
<tr class="row-even"><td>3</td>
<td>RNAME</td>
<td>Node name</td>
</tr>
<tr class="row-odd"><td>4</td>
<td>POS</td>
<td><code class="docutils literal"><span class="pre">1</span></code> if assigned, unused (<code class="docutils literal"><span class="pre">0</span></code>) otherwise</td>
</tr>
<tr class="row-even"><td>5</td>
<td>MAPQ</td>
<td><code class="docutils literal"><span class="pre">60</span></code> if assigned, unused (<code class="docutils literal"><span class="pre">0</span></code>) otherwise</td>
</tr>
<tr class="row-odd"><td>6</td>
<td>CIGAR</td>
<td>Coverage bit-mask (e.g., <cite>7=3X3=</cite> means <cite>1111111000111</cite>) if assigned, unused (<code class="docutils literal"><span class="pre">*</span></code>) otherwise</td>
</tr>
<tr class="row-even"><td>7</td>
<td>RNEXT</td>
<td>Unused (<code class="docutils literal"><span class="pre">*</span></code>)</td>
</tr>
<tr class="row-odd"><td>8</td>
<td>PNEXT</td>
<td>Unused (<code class="docutils literal"><span class="pre">0</span></code>)</td>
</tr>
<tr class="row-even"><td>9</td>
<td>TLEN</td>
<td>Unused (<code class="docutils literal"><span class="pre">0</span></code>)</td>
</tr>
<tr class="row-odd"><td>10</td>
<td>SEQ</td>
<td>Sequence of bases if <code class="docutils literal"><span class="pre">-P</span></code>, unused (<code class="docutils literal"><span class="pre">*</span></code>) otherwise</td>
</tr>
<tr class="row-even"><td>11</td>
<td>QUAL</td>
<td>Base qualities if <code class="docutils literal"><span class="pre">-P</span></code>, unused (<code class="docutils literal"><span class="pre">*</span></code>) otherwise</td>
</tr>
</tbody>
</table>
<table border="1" class="colwidths-given docutils" id="id4">
<caption><span class="caption-text">SAM tags</span><a class="headerlink" href="#id4" title="Permalink to this table">¶</a></caption>
<colgroup>
<col width="17%" />
<col width="17%" />
<col width="67%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Tag</th>
<th class="head">Type</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>h1</td>
<td>integer</td>
<td>Number of shared k-mers</td>
</tr>
<tr class="row-odd"><td>h2</td>
<td>float</td>
<td>Normalized number of shared k-mers</td>
</tr>
<tr class="row-even"><td>hf</td>
<td>float</td>
<td>Proportion of hits</td>
</tr>
<tr class="row-odd"><td>c1</td>
<td>integer</td>
<td>Number of covered positions in the query</td>
</tr>
<tr class="row-even"><td>c2</td>
<td>float</td>
<td>Normalized number of covered positions in the query</td>
</tr>
<tr class="row-odd"><td>cf</td>
<td>float</td>
<td>Proportion of coverage</td>
</tr>
<tr class="row-even"><td>is</td>
<td>int</td>
<td>Number of reported assignments for the query</td>
</tr>
<tr class="row-odd"><td>ii</td>
<td>int</td>
<td>ID of the curent assignment</td>
</tr>
<tr class="row-even"><td>hc</td>
<td>string</td>
<td>Hit bit-mask  (e.g., <cite>7=1X3=</cite> means <cite>11111110111</cite>)</td>
</tr>
</tbody>
</table>
</div></blockquote>
</div>
<div class="section" id="read-assignments-in-a-kraken-like-format">
<h3>Read assignments in a Kraken-like format<a class="headerlink" href="#read-assignments-in-a-kraken-like-format" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://ccb.jhu.edu/software/kraken/MANUAL.html#output-format">Kraken</a></p>
</div>
</div>
<div class="section" id="abundances-estimates-experimental">
<h2>Abundances estimates (experimental)<a class="headerlink" href="#abundances-estimates-experimental" title="Permalink to this headline">¶</a></h2>
<div class="section" id="abundances-in-the-kraken-report-format">
<h3>Abundances in the Kraken report format<a class="headerlink" href="#abundances-in-the-kraken-report-format" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://ccb.jhu.edu/software/kraken/MANUAL.html#sample-reports">kraken-report</a> format:</p>
<blockquote>
<div><table border="1" class="colwidths-given docutils" id="id5">
<caption><span class="caption-text">Kraken report format</span><a class="headerlink" href="#id5" title="Permalink to this table">¶</a></caption>
<colgroup>
<col width="20%" />
<col width="80%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Column</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>1</td>
<td>Percentage of reads covered by the clade rooted at this taxon</td>
</tr>
<tr class="row-odd"><td>2</td>
<td>Number of reads covered by the clade rooted at this taxon</td>
</tr>
<tr class="row-even"><td>3</td>
<td>Number of reads assigned directly to this taxon</td>
</tr>
<tr class="row-odd"><td>4</td>
<td>A rank code, indicating (U)nclassified, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. All other ranks are simply ‘-‘.</td>
</tr>
<tr class="row-even"><td>5</td>
<td>NCBI taxonomy ID</td>
</tr>
<tr class="row-odd"><td>6</td>
<td>Indented scientific name</td>
</tr>
</tbody>
</table>
</div></blockquote>
</div>
<div class="section" id="abundances-in-the-metaphlan2-report-format">
<h3>Abundances in the Metaphlan2 report format<a class="headerlink" href="#abundances-in-the-metaphlan2-report-format" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://bitbucket.org/biobakery/biobakery/wiki/metaphlan2#rst-header-output-files">MetaPhlAn2</a> format</p>
<blockquote>
<div><table border="1" class="colwidths-given docutils" id="id6">
<caption><span class="caption-text">Metaphlan 2 report format</span><a class="headerlink" href="#id6" title="Permalink to this table">¶</a></caption>
<colgroup>
<col width="20%" />
<col width="80%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Column</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>1</td>
<td>Clades, ranging from taxonomic kingdoms (Bacteria, Archaea, etc.) through species</td>
</tr>
<tr class="row-odd"><td>2</td>
<td>The taxonomic level of each clade is prefixed to indicate its level: Kingdom: <code class="docutils literal"><span class="pre">k__</span></code>, Phylum: <code class="docutils literal"><span class="pre">p__</span></code>, Class: <code class="docutils literal"><span class="pre">c__</span></code>, Order: <code class="docutils literal"><span class="pre">o__</span></code>, Family: <code class="docutils literal"><span class="pre">f__</span></code>, Genus: <code class="docutils literal"><span class="pre">g__</span></code>, Species: <code class="docutils literal"><span class="pre">s__</span></code></td>
</tr>
</tbody>
</table>
</div></blockquote>
<p>Since sequence-based profiling is relative and does not provide absolute cellular abundance measures, clades are hierarchically summed. Each level will sum to 100%; that is, the sum of all kindom-level clades is 100%, the sum of all genus-level clades (including unclassified) is also 100%, and so forth. OTU equivalents can be extracted by using only the species-level <code class="docutils literal"><span class="pre">s__</span></code> clades from this file (again, making sure to include clades unclassified at this level).</p>
</div>
<div class="section" id="abundances-in-the-centrifuge-report-format">
<h3>Abundances in the Centrifuge report format<a class="headerlink" href="#abundances-in-the-centrifuge-report-format" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://ccb.jhu.edu/software/centrifuge/manual.shtml#centrifuge-summary-output-the-default-filename-is-centrifuge_report.tsv">Centrifuge</a> format.</p>
<div class="section" id="specification">
<h4>Specification<a class="headerlink" href="#specification" title="Permalink to this headline">¶</a></h4>
<blockquote>
<div><table border="1" class="colwidths-given docutils" id="id7">
<caption><span class="caption-text">Centrifuge format</span><a class="headerlink" href="#id7" title="Permalink to this table">¶</a></caption>
<colgroup>
<col width="20%" />
<col width="80%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Column</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>1</td>
<td>name of a genome, or the name corresponding to a taxonomic ID (the second column) at a rank higher than the strain (e.g., Wigglesworthia glossinidia endosymbiont of Glossina brevipalpis).</td>
</tr>
<tr class="row-odd"><td>2</td>
<td>taxonomic ID (e.g., 36870).</td>
</tr>
<tr class="row-even"><td>3</td>
<td>taxonomic rank (e.g., leaf).</td>
</tr>
<tr class="row-odd"><td>4</td>
<td>number of k-mers propagated up to the node (e.g., 703004).</td>
</tr>
<tr class="row-even"><td>5</td>
<td>number of reads classified to this node including multi-classified reads (divided by the number of assignments, e.g., 5981.37)</td>
</tr>
<tr class="row-odd"><td>6</td>
<td>number of reads uniquely classified to this genomic sequence (e.g., 5964)</td>
</tr>
<tr class="row-even"><td>7</td>
<td>unused</td>
</tr>
</tbody>
</table>
</div></blockquote>
</div>
<div class="section" id="id1">
<h4>Example<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h4>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="c1">#name                                                           taxID   taxRank    kmerCount   numReads   numUniqueReads   abundance</span>
<span class="n">Wigglesworthia</span> <span class="n">glossinidia</span> <span class="n">endosymbiont</span> <span class="n">of</span> <span class="n">Glossina</span> <span class="n">brevipalpis</span> <span class="mi">36870</span>   <span class="n">leaf</span>       <span class="mi">703004</span>      <span class="mf">5981.37</span>    <span class="mi">5964</span>             <span class="mi">0</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="internal-prophyle-formats">
<h2>Internal ProPhyle formats<a class="headerlink" href="#internal-prophyle-formats" title="Permalink to this headline">¶</a></h2>
<div class="section" id="prophyle-index">
<h3>ProPhyle Index<a class="headerlink" href="#prophyle-index" title="Permalink to this headline">¶</a></h3>
<p>Directory with the following files:</p>
<blockquote>
<div><table border="1" class="colwidths-given docutils" id="id8">
<caption><span class="caption-text">ProPhyle index</span><a class="headerlink" href="#id8" title="Permalink to this table">¶</a></caption>
<colgroup>
<col width="20%" />
<col width="80%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">File name</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><code class="docutils literal"><span class="pre">index.fa</span></code></td>
<td>Assembled contigs, name of sequences are of the following format: <code class="docutils literal"><span class="pre">[node_name]&#64;c[contig_id]</span></code></td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">index.fa.amb</span></code></td>
<td>List of ambiguous nucleotides, no values</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">index.fa.ann</span></code></td>
<td>List of contigs and their starting positions in the master string</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">index.fa.[k].klcp</span></code></td>
<td>k-LCP array</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">index.fa.bwt</span></code></td>
<td>Burrows-Wheeler Transform of the master string (merged sequences + reverse completement) + OCC table (BWA format)</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">index.fa.kmers.tsv</span></code></td>
<td>k-mer statistics, format: <code class="docutils literal"><span class="pre">[node_name].[full|reduced].fa</span>&#160;&#160;&#160;&#160;&#160;&#160;&#160; <span class="pre">[#kmers]</span></code>, where <code class="docutils literal"><span class="pre">full</span></code> refers to all associated k-mers and <code class="docutils literal"><span class="pre">reduced</span></code> to represented k-mers</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">index.fa.pac</span></code></td>
<td>Packed sequences (BWA format)</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">index.fa.sa</span></code></td>
<td>Sampled suffix array (BWA format)</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">index.json</span></code></td>
<td>Index parameters: k-mer size (<code class="docutils literal"><span class="pre">k</span></code>), ProPhyle version (<code class="docutils literal"><span class="pre">prophyle-version</span></code>, <code class="docutils literal"><span class="pre">prophyle-revision</span></code>, <code class="docutils literal"><span class="pre">prophyle-commit</span></code>)</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">log.txt</span></code></td>
<td>Log</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">tree.nw</span></code></td>
<td>Phylogenetic tree adjusted for classification</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">tree.preliminary.nw</span></code></td>
<td>Phylogenetic tree before adjusting</td>
</tr>
</tbody>
</table>
</div></blockquote>
</div>
<div class="section" id="compressed-prophyle-index-for-transmission">
<h3>Compressed ProPhyle index for transmission<a class="headerlink" href="#compressed-prophyle-index-for-transmission" title="Permalink to this headline">¶</a></h3>
<p>A <code class="docutils literal"><span class="pre">.tar.gz</span></code> archive with the following subset of the index files:</p>
<blockquote>
<div><table border="1" class="colwidths-given docutils" id="id9">
<caption><span class="caption-text">Compressed ProPhyle index</span><a class="headerlink" href="#id9" title="Permalink to this table">¶</a></caption>
<colgroup>
<col width="20%" />
<col width="80%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">File name</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><code class="docutils literal"><span class="pre">index.fa.amb</span></code></td>
<td>Identical</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">index.fa.ann</span></code></td>
<td>Identical</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">index.fa.bwt</span></code></td>
<td>Burrows-Wheeler Transform <em>without</em> the OCC table (BWA format, before <code class="docutils literal"><span class="pre">bwa</span> <span class="pre">bwtupdate</span></code>)</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">index.fa.kmers.tsv</span></code></td>
<td>Identical</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">index.json</span></code></td>
<td>Identical</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">tree.nw</span></code></td>
<td>Identical</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">tree.preliminary.nw</span></code></td>
<td>Identical</td>
</tr>
</tbody>
</table>
</div></blockquote>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="cli.html" title="Command-line interface"
             >next</a> |</li>
        <li class="right" >
          <a href="compress.html" title="Compressing indexes for transmission"
             >previous</a> |</li>
        <li><a href="index.html">ProPhyle home</a>&#160;|</li>
        <li><a href="contents.html">Documentation</a> &#187;</li>
 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2015-2017, Karel Břinda, Kamil Salikhov, Simone Pignotti, Gregory Kucherov.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.6.5.
    </div>
  </body>
</html>