
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Custom NCBI indexes &#8212; ProPhyle 0.3.1.0 documentation</title>
    <link rel="stylesheet" href="_static/sphinx13.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.3.1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="search" type="application/opensearchdescription+xml"
          title="Search within ProPhyle 0.3.1.0 documentation"
          href="_static/opensearch.xml"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:300,400,700'
          rel='stylesheet' type='text/css' />
 
    <style type="text/css">
      table.right { float: right; margin-left: 20px; }
      table.right td { border: 1px solid #ccc; }
      
    </style>
    <script type="text/javascript">
      // intelligent scrolling of the sidebar content
      $(window).scroll(function() {
        var sb = $('.sphinxsidebarwrapper');
        var win = $(window);
        var sbh = sb.height();
        var offset = $('.sphinxsidebar').position()['top'];
        var wintop = win.scrollTop();
        var winbot = wintop + win.innerHeight();
        var curtop = sb.position()['top'];
        var curbot = curtop + sbh;
        // does sidebar fit in window?
        if (sbh < win.innerHeight()) {
          // yes: easy case -- always keep at the top
          sb.css('top', $u.min([$u.max([0, wintop - offset - 10]),
                                $(document).height() - sbh - 200]));
        } else {
          // no: only scroll if top/bottom edge of sidebar is at
          // top/bottom edge of window
          if (curtop > wintop && curbot > winbot) {
            sb.css('top', $u.max([wintop - offset - 10, 0]));
          } else if (curtop < wintop && curbot < winbot) {
            sb.css('top', $u.min([winbot - sbh - offset - 20,
                                  $(document).height() - sbh - 200]));
          }
        }
      });
    </script>

  </head>
  <body>
<div class="pageheader">
  <ul>
    <li><a href="index.html">Home</a></li>
    <li><a href="install.html">Get it</a></li>
    <li><a href="tutorial.html">Docs</a></li>
    <li><a href="http://github.com/prophyle/prophyle">Extend/Develop</a></li>
  </ul>
  <div>
    <a href="index.html">
      <!--
        <img src="_static/sphinxheader.png" alt="SPHINX" />
      -->
      <div style="color:white;font-size:240%">ProPhyle</div>
      <div style="color:white;font-size:140%">a phylogeny-based metagenomic classifier</div>
    </a>
  </div>
</div>

    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li><a href="index.html">ProPhyle home</a>&#160;|</li>
        <li><a href="contents.html">Documentation</a> &#187;</li>
 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Custom NCBI indexes</a></li>
</ul>

  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/ncbi_db.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="custom-ncbi-indexes">
<span id="ncbi-db"></span><h1>Custom NCBI indexes<a class="headerlink" href="#custom-ncbi-indexes" title="Permalink to this headline">¶</a></h1>
<div class="section" id="download-sequences-from-ncbi">
<h2>Download sequences from NCBI<a class="headerlink" href="#download-sequences-from-ncbi" title="Permalink to this headline">¶</a></h2>
<ul>
<li><p class="first">Create a subdirectory in ProPhyle’s main directory, and change the working directory to the newly created one</p>
</li>
<li><p class="first">Download an <code class="docutils literal"><span class="pre">assembly_summary.txt</span></code> file from NCBI’s FTP server. For RefSeq’s bacterial genomes, follow this <a class="reference external" href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt">link</a>. More information about these files can be found <a class="reference external" href="ftp://ftp.ncbi.nlm.nih.gov/genomes/README_assembly_summary.txt">here</a></p>
</li>
<li><p class="first">Select genomes of interest; the following command will select all complete genomes, but it is sufficient to add more conditions to <code class="docutils literal"><span class="pre">awk</span></code> to select a subset. Fields 1, 6 and 20 contain respectively the accession number, taxonomic identifier and ftp directory of each genome. <code class="docutils literal"><span class="pre">acc2taxid.tsv</span></code> will therefore be a tab separated file containing the taxid corresponding to each sequence’s accession number. Please refer to <a class="reference external" href="https://www.ncbi.nlm.nih.gov/genome/doc/ftpfaq">NCBI’s genomes download FAQ</a> for further information):</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">awk</span> <span class="o">-</span><span class="n">F</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span> <span class="o">-</span><span class="n">v</span> <span class="n">OFS</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span> <span class="s1">&#39;$12==&quot;Complete Genome&quot; &amp;&amp; $11==&quot;latest&quot;</span><span class="se">\</span>
<span class="s1">        {print $1, $6, $20}&#39;</span> <span class="n">assembly_summary</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span><span class="n">ftpselection</span><span class="o">.</span><span class="n">tsv</span>
<span class="n">cut</span> <span class="o">-</span><span class="n">f</span> <span class="mi">3</span> <span class="n">ftpselection</span><span class="o">.</span><span class="n">tsv</span> <span class="o">|</span> <span class="n">awk</span> <span class="s1">&#39;BEGIN{FS=OFS=&quot;/&quot;;filesuffix=&quot;genomic.fna.gz&quot;}</span><span class="se">\</span>
<span class="s1">        {ftpdir=$0;asm=$10;file=asm&quot;_&quot;filesuffix;print ftpdir,file}&#39;</span> <span class="o">&gt;</span><span class="n">ftpfilepaths</span><span class="o">.</span><span class="n">tsv</span>
<span class="n">cut</span> <span class="o">-</span><span class="n">f</span> <span class="mi">1</span><span class="p">,</span><span class="mi">2</span> <span class="n">ftpselection</span><span class="o">.</span><span class="n">tsv</span> <span class="o">&gt;</span><span class="n">acc2taxid</span><span class="o">.</span><span class="n">tsv</span>
</pre></div>
</div>
</li>
<li><p class="first">Download selected sequences using <code class="docutils literal"><span class="pre">parallel</span> <span class="pre">--gnu</span> <span class="pre">-j</span> <span class="pre">24</span> <span class="pre">-a</span> <span class="pre">ftpfilepaths.tsv</span> <span class="pre">wget</span></code>. Number of jobs (<code class="docutils literal"><span class="pre">-j</span></code> option) should be changed according to the number of      available cores and bandwidth.</p>
</li>
<li><p class="first">Extract them using <code class="docutils literal"><span class="pre">parallel</span> <span class="pre">--gnu</span> <span class="pre">-j</span> <span class="pre">24</span> <span class="pre">-a</span> <span class="pre">ftpfilepaths.tsv</span> <span class="pre">gunzip</span> <span class="pre">{/}</span></code></p>
</li>
<li><p class="first">Create a <a class="reference external" href="http://www.htslib.org/doc/faidx.html">fasta index</a> for each file using <code class="docutils literal"><span class="pre">find</span> <span class="pre">.</span> <span class="pre">-name</span> <span class="pre">'*.fna'</span> <span class="pre">|</span> <span class="pre">parallel</span> <span class="pre">--no-notice</span> <span class="pre">--verbose</span> <span class="pre">samtools</span> <span class="pre">faidx</span> <span class="pre">{}</span></code></p>
</li>
</ul>
</div>
<div class="section" id="build-a-tree">
<h2>Build a tree<a class="headerlink" href="#build-a-tree" title="Permalink to this headline">¶</a></h2>
<p>Build a taxonomic tree for the downloaded sequences using:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">prophyle_homedir</span><span class="o">/</span><span class="n">prophyle</span><span class="o">/</span><span class="n">prophyle_ncbitree</span><span class="o">.</span><span class="n">py</span> <span class="o">&lt;</span><span class="n">library_name</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">library_main_dir</span><span class="o">&gt;</span>\
              <span class="o">&lt;</span><span class="n">output_file</span><span class="o">&gt;</span> <span class="n">acc2taxid</span><span class="o">.</span><span class="n">tsv</span> <span class="o">-</span><span class="n">l</span> <span class="o">&lt;</span><span class="n">log_file</span><span class="o">&gt;</span>
</pre></div>
</div>
<p>Taxonomic identifiers are assigned to the sequences first, and then the tree is
built using <a class="reference external" href="http://etetoolkit.org/">ETE Toolkit</a> and saved with newick format
1. Necessary node attributes are:</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">name</span></code>: unique node name (ideally the TaxID of the node)</li>
<li><code class="docutils literal"><span class="pre">taxid</span></code>: unique taxonomic identifier</li>
<li><code class="docutils literal"><span class="pre">seqname</span></code>: names of the sequences sharing the same taxid, separated by &#64;</li>
<li><code class="docutils literal"><span class="pre">fastapath</span></code>: paths of the sequences’ fasta files, separated by &#64; (relative paths from ProPhyle’s home directory)</li>
<li><code class="docutils literal"><span class="pre">infasta_offset</span></code>: positions where each sequence starts inside the corresponding fasta files, separated by &#64;</li>
<li><code class="docutils literal"><span class="pre">base_len</span></code>: length of each sequence, separated by &#64;</li>
</ul>
<p><code class="docutils literal"><span class="pre">seqname</span></code>, <code class="docutils literal"><span class="pre">infasta_offset</span></code> and <code class="docutils literal"><span class="pre">base_len</span></code> can be found in samtools’
<a class="reference external" href="http://www.htslib.org/doc/faidx.html">fasta index</a>. Other optional attributes
are <code class="docutils literal"><span class="pre">sci_name</span></code>, <code class="docutils literal"><span class="pre">named_lineage</span></code>, <code class="docutils literal"><span class="pre">lineage</span></code>, <code class="docutils literal"><span class="pre">rank</span></code> (more info
<a class="reference external" href="http://etetoolkit.org/docs/latest/tutorial/tutorial_ncbitaxonomy.html#automatic-tree-annotation-using-ncbi-taxonomy">here</a>
). Using ETE library or modifying the <code class="docutils literal"><span class="pre">prophyle_ncbitree.py</span></code> script it is
easy to adapt any tree to match the requirements above.</p>
<p>If taxonomic information is not available, <code class="docutils literal"><span class="pre">taxid</span></code> attribute can be replaced by the node name. This could be useful to perform classification over custom phylogenetic trees.</p>
</div>
<div class="section" id="build-the-index">
<h2>Build the index<a class="headerlink" href="#build-the-index" title="Permalink to this headline">¶</a></h2>
<p>Run the standard command to build ProPhyle index:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">prophyle</span> <span class="n">index</span> <span class="o">-</span><span class="n">k</span> <span class="o">&lt;</span><span class="n">kmer_length</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">tree_1</span><span class="o">.</span><span class="n">nw</span><span class="o">&gt;</span> <span class="p">[</span><span class="o">&lt;</span><span class="n">tree_2</span><span class="o">.</span><span class="n">nw</span><span class="o">&gt;</span> <span class="o">...</span><span class="p">]</span> <span class="o">&lt;</span><span class="n">index_dir</span><span class="o">&gt;</span>
</pre></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li><a href="index.html">ProPhyle home</a>&#160;|</li>
        <li><a href="contents.html">Documentation</a> &#187;</li>
 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2017, Karel Břinda, Kamil Salikhov, Simone Pignotti, Gregory Kucherov.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.6.5.
    </div>
  </body>
</html>