
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  
  <!-- Licensed under the Apache 2.0 License -->
  <link rel="stylesheet" type="text/css" href="../_static/fonts/open-sans/stylesheet.css" />
  <!-- Licensed under the SIL Open Font License -->
  <link rel="stylesheet" type="text/css" href="../_static/fonts/source-serif-pro/source-serif-pro.css" />
  <link rel="stylesheet" type="text/css" href="../_static/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="../_static/css/bootstrap-theme.min.css" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
    <title>Custom NCBI indexes &#8212; ProPhyle  documentation</title>
    <link rel="stylesheet" href="../_static/guzzle.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../.static/prophyle.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Reference" href="../reference/index.html" />
    <link rel="prev" title="Default databases" href="default_db.html" />
  
   

  </head>
  <body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="../reference/index.html" title="Reference"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="default_db.html" title="Default databases"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">ProPhyle  documentation</a> &#187;</li>
          <li class="nav-item nav-item-1"><a href="index.html" accesskey="U">Tutorials</a> &#187;</li> 
      </ul>
    </div>
    <div class="container-wrapper">

      <div id="mobile-toggle">
        <a href="#"><span class="glyphicon glyphicon-align-justify" aria-hidden="true"></span></a>
      </div>
  <div id="left-column">
    <div class="sphinxsidebar">
        <a href="
    ../index.html" class="text-logo">ProPhyle</a>
        
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <h2>Table Of Contents</h2>
  </div>
  <div class="sidebar-toc">
    
    
      <p class="caption"><span class="caption-text">ProPhyle Manual</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../introduction.html">Introduction</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="index.html">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="install.html">Installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="default_db.html">Default databases</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Custom NCBI indexes</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../reference/index.html">Reference</a></li>
</ul>

    
  </div>
</div>
        
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <div id="main-search">
      <form class="form-inline" action="../search.html" method="GET" role="form">
        <div class="input-group">
          <input name="q" type="text" class="form-control" placeholder="Search...">
        </div>
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
      </form>
    </div>
  </div>
</div>
    </div>
  </div>
        <div id="right-column">
          
          <div role="navigation" aria-label="breadcrumbs navigation">
            <ol class="breadcrumb">
              <li><a href="../index.html">Docs</a></li>
              
                <li><a href="index.html">Tutorials</a></li>
              
              <li>Custom NCBI indexes</li>
            </ol>
          </div>
          
          <div class="document clearer body">
            
  <div class="section" id="custom-ncbi-indexes">
<span id="ncbi-db"></span><h1>Custom NCBI indexes<a class="headerlink" href="#custom-ncbi-indexes" title="Permalink to this headline">¶</a></h1>
<div class="section" id="download-sequences-from-ncbi">
<h2>Download sequences from NCBI<a class="headerlink" href="#download-sequences-from-ncbi" title="Permalink to this headline">¶</a></h2>
<ul>
<li><p class="first">Create a subdirectory in ProPhyle’s main directory, and change the working directory to the newly created one</p>
</li>
<li><p class="first">Download an <code class="docutils literal"><span class="pre">assembly_summary.txt</span></code> file from NCBI’s FTP server. For RefSeq’s bacterial genomes, follow this <a class="reference external" href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt">link</a>. More information about these files can be found <a class="reference external" href="ftp://ftp.ncbi.nlm.nih.gov/genomes/README_assembly_summary.txt">here</a></p>
</li>
<li><p class="first">Select genomes of interest; the following command will select all complete genomes, but it is sufficient to add more conditions to <code class="docutils literal"><span class="pre">awk</span></code> to select a subset. Fields 1, 6 and 20 contain respectively the accession number, taxonomic identifier and ftp directory of each genome. <code class="docutils literal"><span class="pre">acc2taxid.tsv</span></code> will therefore be a tab separated file containing the taxid corresponding to each sequence’s accession number. Please refer to <a class="reference external" href="https://www.ncbi.nlm.nih.gov/genome/doc/ftpfaq">NCBI’s genomes download FAQ</a> for further information):</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">awk</span> <span class="o">-</span><span class="n">F</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span> <span class="o">-</span><span class="n">v</span> <span class="n">OFS</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span> <span class="s1">&#39;$12==&quot;Complete Genome&quot; &amp;&amp; $11==&quot;latest&quot;</span><span class="se">\</span>
<span class="s1">        {print $1, $6, $20}&#39;</span> <span class="n">assembly_summary</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span><span class="n">ftpselection</span><span class="o">.</span><span class="n">tsv</span>
<span class="n">cut</span> <span class="o">-</span><span class="n">f</span> <span class="mi">3</span> <span class="n">ftpselection</span><span class="o">.</span><span class="n">tsv</span> <span class="o">|</span> <span class="n">awk</span> <span class="s1">&#39;BEGIN{FS=OFS=&quot;/&quot;;filesuffix=&quot;genomic.fna.gz&quot;}</span><span class="se">\</span>
<span class="s1">        {ftpdir=$0;asm=$10;file=asm&quot;_&quot;filesuffix;print ftpdir,file}&#39;</span> <span class="o">&gt;</span><span class="n">ftpfilepaths</span><span class="o">.</span><span class="n">tsv</span>
<span class="n">cut</span> <span class="o">-</span><span class="n">f</span> <span class="mi">1</span><span class="p">,</span><span class="mi">2</span> <span class="n">ftpselection</span><span class="o">.</span><span class="n">tsv</span> <span class="o">&gt;</span><span class="n">acc2taxid</span><span class="o">.</span><span class="n">tsv</span>
</pre></div>
</div>
</li>
<li><p class="first">Download selected sequences using <code class="docutils literal"><span class="pre">parallel</span> <span class="pre">--gnu</span> <span class="pre">-j</span> <span class="pre">24</span> <span class="pre">-a</span> <span class="pre">ftpfilepaths.tsv</span> <span class="pre">wget</span></code>. Number of jobs (<code class="docutils literal"><span class="pre">-j</span></code> option) should be changed according to the number of      available cores and bandwidth.</p>
</li>
<li><p class="first">Extract them using <code class="docutils literal"><span class="pre">parallel</span> <span class="pre">--gnu</span> <span class="pre">-j</span> <span class="pre">24</span> <span class="pre">-a</span> <span class="pre">ftpfilepaths.tsv</span> <span class="pre">gunzip</span> <span class="pre">{/}</span></code></p>
</li>
<li><p class="first">Create a <a class="reference external" href="http://www.htslib.org/doc/faidx.html">fasta index</a> for each file using <code class="docutils literal"><span class="pre">find</span> <span class="pre">.</span> <span class="pre">-name</span> <span class="pre">'*.fna'</span> <span class="pre">|</span> <span class="pre">parallel</span> <span class="pre">--no-notice</span> <span class="pre">--verbose</span> <span class="pre">samtools</span> <span class="pre">faidx</span> <span class="pre">{}</span></code></p>
</li>
</ul>
</div>
<div class="section" id="build-a-tree">
<h2>Build a tree<a class="headerlink" href="#build-a-tree" title="Permalink to this headline">¶</a></h2>
<p>Build a taxonomic tree for the downloaded sequences using:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">prophyle_homedir</span><span class="o">/</span><span class="n">prophyle</span><span class="o">/</span><span class="n">prophyle_ncbitree</span><span class="o">.</span><span class="n">py</span> <span class="o">&lt;</span><span class="n">library_name</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">library_main_dir</span><span class="o">&gt;</span>\
              <span class="o">&lt;</span><span class="n">output_file</span><span class="o">&gt;</span> <span class="n">acc2taxid</span><span class="o">.</span><span class="n">tsv</span> <span class="o">-</span><span class="n">l</span> <span class="o">&lt;</span><span class="n">log_file</span><span class="o">&gt;</span>
</pre></div>
</div>
<p>Taxonomic identifiers are assigned to the sequences first, and then the tree is
built using <a class="reference external" href="http://etetoolkit.org/">ETE Toolkit</a> and saved with newick format
1. Necessary node attributes are:</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">name</span></code>: unique node name (ideally the TaxID of the node)</li>
<li><code class="docutils literal"><span class="pre">taxid</span></code>: unique taxonomic identifier</li>
<li><code class="docutils literal"><span class="pre">seqname</span></code>: names of the sequences sharing the same taxid, separated by &#64;</li>
<li><code class="docutils literal"><span class="pre">fastapath</span></code>: paths of the sequences’ fasta files, separated by &#64; (relative paths from ProPhyle’s home directory)</li>
<li><code class="docutils literal"><span class="pre">infasta_offset</span></code>: positions where each sequence starts inside the corresponding fasta files, separated by &#64;</li>
<li><code class="docutils literal"><span class="pre">base_len</span></code>: length of each sequence, separated by &#64;</li>
</ul>
<p><code class="docutils literal"><span class="pre">seqname</span></code>, <code class="docutils literal"><span class="pre">infasta_offset</span></code> and <code class="docutils literal"><span class="pre">base_len</span></code> can be found in samtools’
<a class="reference external" href="http://www.htslib.org/doc/faidx.html">fasta index</a>. Other optional attributes
are <code class="docutils literal"><span class="pre">sci_name</span></code>, <code class="docutils literal"><span class="pre">named_lineage</span></code>, <code class="docutils literal"><span class="pre">lineage</span></code>, <code class="docutils literal"><span class="pre">rank</span></code> (more info
<a class="reference external" href="http://etetoolkit.org/docs/latest/tutorial/tutorial_ncbitaxonomy.html#automatic-tree-annotation-using-ncbi-taxonomy">here</a>
). Using ETE library or modifying the <code class="docutils literal"><span class="pre">prophyle_ncbitree.py</span></code> script it is
easy to adapt any tree to match the requirements above.</p>
<p>If taxonomic information is not available, <code class="docutils literal"><span class="pre">taxid</span></code> attribute can be replaced by the node name. This could be useful to perform classification over custom phylogenetic trees.</p>
</div>
<div class="section" id="build-the-index">
<h2>Build the index<a class="headerlink" href="#build-the-index" title="Permalink to this headline">¶</a></h2>
<p>Run the standard command to build ProPhyle index:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">prophyle</span> <span class="n">index</span> <span class="o">-</span><span class="n">k</span> <span class="o">&lt;</span><span class="n">kmer_length</span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">tree_1</span><span class="o">.</span><span class="n">nw</span><span class="o">&gt;</span> <span class="p">[</span><span class="o">&lt;</span><span class="n">tree_2</span><span class="o">.</span><span class="n">nw</span><span class="o">&gt;</span> <span class="o">...</span><span class="p">]</span> <span class="o">&lt;</span><span class="n">index_dir</span><span class="o">&gt;</span>
</pre></div>
</div>
</div>
</div>


          </div>
            
  <div class="footer-relations">
    
      <div class="pull-left">
        <a class="btn btn-default" href="default_db.html" title="previous chapter (use the left arrow)">Default databases</a>
      </div>
    
      <div class="pull-right">
        <a class="btn btn-default" href="../reference/index.html" title="next chapter (use the right arrow)">Reference</a>
      </div>
    </div>
    <div class="clearer"></div>
  
        </div>
        <div class="clearfix"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="../reference/index.html" title="Reference"
             >next</a> |</li>
        <li class="right" >
          <a href="default_db.html" title="Default databases"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">ProPhyle  documentation</a> &#187;</li>
          <li class="nav-item nav-item-1"><a href="index.html" >Tutorials</a> &#187;</li> 
      </ul>
    </div>
<script type="text/javascript">
  $("#mobile-toggle a").click(function () {
    $("#left-column").toggle();
  });
</script>
<script type="text/javascript" src="../_static/js/bootstrap.js"></script>
  <div class="footer">
    &copy; Copyright 2017, Karel Břinda, Kamil Salikhov, Simone Pignotti, Gregory Kucherov. Created using <a href="http://sphinx.pocoo.org/">Sphinx</a>.
  </div>
  </body>
</html>