
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  
  <!-- Licensed under the Apache 2.0 License -->
  <link rel="stylesheet" type="text/css" href="../_static/fonts/open-sans/stylesheet.css" />
  <!-- Licensed under the SIL Open Font License -->
  <link rel="stylesheet" type="text/css" href="../_static/fonts/source-serif-pro/source-serif-pro.css" />
  <link rel="stylesheet" type="text/css" href="../_static/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="../_static/css/bootstrap-theme.min.css" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
    <title>Reference &#8212; ProPhyle  documentation</title>
    <link rel="stylesheet" href="../_static/guzzle.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../.static/prophyle.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="prev" title="Custom NCBI indexes" href="../tutorials/ncbi_db.html" />
  
   

  </head>
  <body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="../tutorials/ncbi_db.html" title="Custom NCBI indexes"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">ProPhyle  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="container-wrapper">

      <div id="mobile-toggle">
        <a href="#"><span class="glyphicon glyphicon-align-justify" aria-hidden="true"></span></a>
      </div>
  <div id="left-column">
    <div class="sphinxsidebar">
        <a href="
    ../index.html" class="text-logo">ProPhyle</a>
        
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <h2>Table Of Contents</h2>
  </div>
  <div class="sidebar-toc">
    
    
      <p class="caption"><span class="caption-text">ProPhyle Manual</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Reference</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#formats">Formats</a></li>
<li class="toctree-l2"><a class="reference internal" href="#main-program-s-reference">Main program’s reference</a></li>
<li class="toctree-l2"><a class="reference internal" href="#other-programs-reference">Other programs’ reference</a></li>
</ul>
</li>
</ul>

    
  </div>
</div>
        
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <div id="main-search">
      <form class="form-inline" action="../search.html" method="GET" role="form">
        <div class="input-group">
          <input name="q" type="text" class="form-control" placeholder="Search...">
        </div>
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
      </form>
    </div>
  </div>
</div>
    </div>
  </div>
        <div id="right-column">
          
          <div role="navigation" aria-label="breadcrumbs navigation">
            <ol class="breadcrumb">
              <li><a href="../index.html">Docs</a></li>
              
              <li>Reference</li>
            </ol>
          </div>
          
          <div class="document clearer body">
            
  <div class="section" id="reference">
<span id="id1"></span><h1>Reference<a class="headerlink" href="#reference" title="Permalink to this headline">¶</a></h1>
<p>Here you can find format specifications and an automatically generated reference
for ProPhyle’s CLI.</p>
<div class="section" id="formats">
<h2>Formats<a class="headerlink" href="#formats" title="Permalink to this headline">¶</a></h2>
<div class="section" id="trees">
<h3>Trees<a class="headerlink" href="#trees" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="http://evolution.genetics.washington.edu/phylip/newicktree.html">Newick</a>
format <a class="reference external" href="http://etetoolkit.org/docs/latest/tutorial/tutorial_trees.html#reading-and-writing-newick-trees">1</a>
with NHX annotations, which can be easily created and modified using the
<a class="reference external" href="http://etetoolkit.org/">ete3</a> python package.</p>
</div>
<div class="section" id="classification-output">
<h3>Classification output<a class="headerlink" href="#classification-output" title="Permalink to this headline">¶</a></h3>
<p>Support for both <a class="reference external" href="http://samtools.github.io/hts-specs/">SAM</a> and <a class="reference external" href="https://ccb.jhu.edu/software/kraken/MANUAL.html#output-format">Kraken</a>
output formats.</p>
</div>
<div class="section" id="analysis-output">
<h3>Analysis output<a class="headerlink" href="#analysis-output" title="Permalink to this headline">¶</a></h3>
<ul>
<li><dl class="first docutils">
<dt><a class="reference external" href="https://ccb.jhu.edu/software/kraken/MANUAL.html#sample-reports">kraken-report</a> format:</dt>
<dd><ol class="first last arabic simple">
<li>Percentage of reads covered by the clade rooted at this taxon</li>
<li>Number of reads covered by the clade rooted at this taxon</li>
<li>Number of reads assigned directly to this taxon</li>
<li>A rank code, indicating (U)nclassified, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. All other ranks are simply ‘-‘.</li>
<li>NCBI taxonomy ID</li>
<li>indented scientific name</li>
</ol>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt><a class="reference external" href="https://bitbucket.org/biobakery/biobakery/wiki/metaphlan2#rst-header-output-files">MetaPhlAn2</a> format:</dt>
<dd><p class="first last">1. clades, ranging from taxonomic kingdoms (Bacteria, Archaea, etc.) through species.
The taxonomic level of each clade is prefixed to indicate its level: Kingdom: <code class="docutils literal"><span class="pre">k__</span></code>, Phylum: <code class="docutils literal"><span class="pre">p__</span></code>, Class: <code class="docutils literal"><span class="pre">c__</span></code>, Order: <code class="docutils literal"><span class="pre">o__</span></code>, Family: <code class="docutils literal"><span class="pre">f__</span></code>, Genus: <code class="docutils literal"><span class="pre">g__</span></code>, Species: <code class="docutils literal"><span class="pre">s__</span></code>.
Since sequence-based profiling is relative and does not provide absolute cellular abundance measures,
clades are hierarchically summed. Each level will sum to 100%; that is, the sum of all kindom-level
clades is 100%, the sum of all genus-level clades (including unclassified) is also 100%, and so forth.
OTU equivalents can be extracted by using only the species-level <code class="docutils literal"><span class="pre">s__</span></code> clades from this file
(again, making sure to include clades unclassified at this level).</p>
</dd>
</dl>
</li>
<li><p class="first">Custom <a class="reference external" href="https://ccb.jhu.edu/software/centrifuge/manual.shtml#centrifuge-summary-output-the-default-filename-is-centrifuge_report.tsv">Centrifuge</a> format:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="c1">#name                                                           taxID   taxRank    kmerCount   numReads   numUniqueReads   abundance</span>
<span class="n">Wigglesworthia</span> <span class="n">glossinidia</span> <span class="n">endosymbiont</span> <span class="n">of</span> <span class="n">Glossina</span> <span class="n">brevipalpis</span> <span class="mi">36870</span>   <span class="n">leaf</span>       <span class="mi">703004</span>      <span class="mf">5981.37</span>    <span class="mi">5964</span>             <span class="mi">0</span>
</pre></div>
</div>
</li>
</ul>
<ol class="arabic simple">
<li>name of a genome, or the name corresponding to a taxonomic ID (the second column) at a rank higher than the strain (e.g., Wigglesworthia glossinidia endosymbiont of Glossina brevipalpis).</li>
<li>taxonomic ID (e.g., 36870).</li>
<li>taxonomic rank (e.g., leaf).</li>
<li>number of k-mers propagated up to the node (e.g., 703004).</li>
<li>number of reads classified to this node including multi-classified reads (divided by the number of assignments, e.g., 5981.37).</li>
<li>number of reads uniquely classified to this genomic sequence (e.g., 5964).</li>
<li>not used yet.</li>
</ol>
</div>
</div>
<div class="section" id="main-program-s-reference">
<h2>Main program’s reference<a class="headerlink" href="#main-program-s-reference" title="Permalink to this headline">¶</a></h2>
<div class="section" id="prophyle-list-of-subcommands">
<h3><code class="docutils literal"><span class="pre">prophyle</span></code> (list of subcommands)<a class="headerlink" href="#prophyle-list-of-subcommands" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle  -h

usage: prophyle.py [-h] [-v]  ...

Program: prophyle (phylogeny-based metagenomic classification)
Version: 0.2.0.0
Authors: Karel Brinda &lt;kbrinda@hsph.harvard.edu&gt;, Kamil Salikhov &lt;kamil.salikhov@univ-mlv.fr&gt;,
                 Simone Pignotti &lt;pignottisimone@gmail.com&gt;, Gregory Kucherov &lt;gregory.kucherov@univ-mlv.fr&gt;

Usage:   prophyle &lt;command&gt; [options]

optional arguments:
  -h, --help     show this help message and exit
  -v, --version  show program&#39;s version number and exit

subcommands:

    download     download a genomic database
    index        build index
    classify     classify reads
    analyze      analyze results
    compress     compress a ProPhyle index (experimental)
    decompress   decompress a compressed ProPhyle index (experimental)
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-download">
<h3><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">download</span></code><a class="headerlink" href="#prophyle-download" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle download -h

usage: prophyle.py download [-h] [-d DIR] [-l STR] [-F]
                            &lt;library&gt; [&lt;library&gt; ...]

positional arguments:
  &lt;library&gt;   genomic library [&#39;bacteria&#39;, &#39;viruses&#39;, &#39;plasmids&#39;, &#39;hmp&#39;,
              &#39;all&#39;]

optional arguments:
  -h, --help  show this help message and exit
  -d DIR      directory for the tree and the sequences [~/prophyle]
  -l STR      log file
  -F          rewrite library files if they already exist
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-index">
<h3><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">index</span></code><a class="headerlink" href="#prophyle-index" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle index -h

usage: prophyle.py index [-h] [-g DIR] [-j INT] [-k INT] [-l STR] [-s FLOAT]
                         [-F] [-M] [-P] [-K] [-T] [-A]
                         &lt;tree.nw&gt; [&lt;tree.nw&gt; ...] &lt;index.dir&gt;

positional arguments:
  &lt;tree.nw&gt;    phylogenetic tree (in Newick/NHX)
  &lt;index.dir&gt;  index directory (will be created)

optional arguments:
  -h, --help   show this help message and exit
  -g DIR       directory with the library sequences [dir. of the first tree]
  -j INT       number of threads [auto (4)]
  -k INT       k-mer length [31]
  -l STR       log file [&lt;index.dir&gt;/log.txt]
  -s FLOAT     rate of sampling of the tree [no sampling]
  -F           rewrite index files if they already exist
  -M           mask repeats/low complexity regions (using DustMasker)
  -P           do not add prefixes to node names when multiple trees are used
  -K           skip k-LCP construction (then restarted search only)
  -T           keep temporary files from k-mer propagation
  -A           autocomplete tree (names of internal nodes and FASTA paths)
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-classify">
<h3><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">classify</span></code><a class="headerlink" href="#prophyle-classify" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle classify -h

usage: prophyle.py classify [-h] [-k INT] [-K] [-m {h1,c1}] [-f {kraken,sam}]
                            [-l STR] [-A] [-L] [-P] [-C]
                            &lt;index.dir&gt; &lt;reads1.fq&gt; [&lt;reads2.fq&gt;]

positional arguments:
  &lt;index.dir&gt;      index directory
  &lt;reads1.fq&gt;      first file with reads in FASTA/FASTQ (- for standard input)
  &lt;reads2.fq&gt;      second file with reads in FASTA/FASTQ

optional arguments:
  -h, --help       show this help message and exit
  -k INT           k-mer length [detect automatically from the index]
  -K               use restarted search for matching rather than rolling
                   window (slower, but k-LCP is not needed)
  -m {h1,c1}       measure: h1=hit count, c1=coverage [h1]
  -f {kraken,sam}  output format [sam]
  -l STR           log file
  -A               annotate assignments (using tax. information from NHX)
  -L               use LCA when tie (multiple hits with the same score)
  -P               incorporate sequences and qualities into SAM records
  -C               use C++ impl. of the assignment algorithm (experimental)
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-analyze">
<h3><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">analyze</span></code><a class="headerlink" href="#prophyle-analyze" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle analyze -h

usage: prophyle.py analyze [-h] [-s [&#39;w&#39;, &#39;u&#39;, &#39;wl&#39;, &#39;ul&#39;]]
                           [-f [&#39;sam&#39;, &#39;bam&#39;, &#39;cram&#39;, &#39;uncompressed_bam&#39;, &#39;kraken&#39;, &#39;histo&#39;]]
                           {index_dir, tree.nw} &lt;out.pref&gt; &lt;classified.bam&gt;
                           [&lt;classified.bam&gt; ...]

positional arguments:
  {index_dir, tree.nw}     index directory or phylogenetic tree
  &lt;out.pref&gt;               output prefix
  &lt;classified.bam&gt;         classified reads (use &#39;-&#39; for stdin)

optional arguments:
  -h, --help               show this help message and exit
  -s [&#39;w&#39;, &#39;u&#39;, &#39;wl&#39;, &#39;ul&#39;]
                           statistics to use for the computation of
                           histograms: w (default) =&gt; weighted assignments; u
                           =&gt; unique assignments, non-weighted; wl =&gt; weighted
                           assignments, propagated to leaves; ul =&gt; unique
                           assignments, propagated to leaves.
  -f [&#39;sam&#39;, &#39;bam&#39;, &#39;cram&#39;, &#39;uncompressed_bam&#39;, &#39;kraken&#39;, &#39;histo&#39;]
                           Input format of assignments [auto]
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-compress">
<h3><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">compress</span></code><a class="headerlink" href="#prophyle-compress" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle compress -h

usage: prophyle.py compress [-h] &lt;index.dir&gt; [&lt;archive.tar.gz&gt;]

positional arguments:
  &lt;index.dir&gt;       index directory
  &lt;archive.tar.gz&gt;  output archive [&lt;index.dir&gt;.tar.gz]

optional arguments:
  -h, --help        show this help message and exit
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-decompress">
<h3><code class="docutils literal"><span class="pre">prophyle</span> <span class="pre">decompress</span></code><a class="headerlink" href="#prophyle-decompress" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle decompress -h

usage: prophyle.py decompress [-h] [-K] archive.tar.gz [output.dir]

positional arguments:
  archive.tar.gz  output archive
  output.dir      output directory [./]

optional arguments:
  -h, --help      show this help message and exit
  -K              skip k-LCP construction
</pre></div>
</div>
</div></blockquote>
</div>
</div>
<div class="section" id="other-programs-reference">
<h2>Other programs’ reference<a class="headerlink" href="#other-programs-reference" title="Permalink to this headline">¶</a></h2>
<div class="section" id="prophyle-ncbi-tree">
<h3><code class="docutils literal"><span class="pre">prophyle_ncbi_tree</span></code><a class="headerlink" href="#prophyle-ncbi-tree" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_ncbi_tree.py -h

usage: prophyle_ncbi_tree.py [-h] [-l log_file] [-r red_factor] [-u root]
                             &lt;library&gt; &lt;library_dir&gt; &lt;output_file&gt; &lt;taxid_map&gt;

Program: prophyle_ncbi_tree Build a taxonomic tree in the New Hampshire newick
format #1 for NCBI sequences

positional arguments:
  &lt;library&gt;      directory with the library sequences (e.g. bacteria, viruses
                 etc.)
  &lt;library_dir&gt;  library path (parent of library, e.g. main ProPhyle
                 directory)
  &lt;output_file&gt;  output file
  &lt;taxid_map&gt;    tab separated accession number to taxid map

optional arguments:
  -h, --help     show this help message and exit
  -l log_file    log file [stderr]
  -r red_factor  build reduced tree (one sequence every n)
  -u root        root of the tree (e.g. Bacteria); will exclude sequences
                 which are not its descendants
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-assembler">
<h3><code class="docutils literal"><span class="pre">prophyle_assembler</span></code><a class="headerlink" href="#prophyle-assembler" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_assembler -h


Program:  prophyle_assembler (greedy assembler for ProPhyle)
Contact:  Karel Brinda &lt;karel.brinda@gmail.com&gt;

Usage:    prophyle_assembler [options]

Examples: prophyle_assembler -k 15 -i f1.fa -i f2.fa -x fx.fa
             - compute intersection of f1 and f2
          prophyle_assembler -k 15 -i f1.fa -i f2.fa -x fx.fa -o g1.fa -o g2.fa
             - compute intersection of f1 and f2, and subtract it from them
          prophyle_assembler -k 15 -i f1.fa -o g1.fa
             - re-assemble f1 to g1

Command-line parameters:
 -k INT   K-mer size.
 -i FILE  Input FASTA file (can be used multiple times).
 -o FILE  Output FASTA file (if used, must be used as many times as -i).
 -x FILE  Compute intersection, subtract it, save it.
 -s FILE  Output file with k-mer statistics.
 -S       Silent mode.

Note that &#39;-&#39; can be used for standard input/output.
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-index-list-of-subcommands">
<h3><code class="docutils literal"><span class="pre">prophyle_index</span></code> (list of subcommands)<a class="headerlink" href="#prophyle-index-list-of-subcommands" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_index -h


Program: prophyle_index (alignment of k-mers)
Contact: Kamil Salikhov &lt;kamil.salikhov@univ-mlv.fr&gt;

Usage:   prophyle_index command [options]

Command: build     construct index
         query     query reads against index
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="prophyle-index-build">
<h3><code class="docutils literal"><span class="pre">prophyle_index</span> <span class="pre">build</span></code><a class="headerlink" href="#prophyle-index-build" title="Permalink to this headline">¶</a></h3>
<div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_index build -h


Usage:   prophyle_index build &lt;prefix&gt;

Options: -k INT    length of k-mer
         -s        construct k-LCP and SA in parallel
         -i        sampling distance for SA
</pre></div>
</div>
</div>
<div class="section" id="prophyle-index-query">
<h3><code class="docutils literal"><span class="pre">prophyle_index</span> <span class="pre">query</span></code><a class="headerlink" href="#prophyle-index-query" title="Permalink to this headline">¶</a></h3>
<div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_index query -h


Usage:   prophyle_index query [options] &lt;prefix&gt; &lt;in.fq&gt;

Options: -k INT    length of k-mer
         -u        use k-LCP for querying
         -v        output set of chromosomes for every k-mer
         -p        do not check whether k-mer is on border of two contigs, and show such k-mers in output
         -b        print sequences and base qualities
         -l STR    log file name to output statistics
         -t INT    number of threads [1]
</pre></div>
</div>
</div>
<div class="section" id="prophyle-assignment">
<h3><code class="docutils literal"><span class="pre">prophyle_assignment</span></code><a class="headerlink" href="#prophyle-assignment" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_assignment.py -h

usage: prophyle_assignment.py [-h] [-f {kraken,sam}] [-m {h1,c1}] [-A] [-L]
                              [-X] [-D]
                              &lt;tree.nhx&gt; &lt;k&gt; &lt;assignments.txt&gt;

Implementation of assignment algorithm

positional arguments:
  &lt;tree.nhx&gt;         phylogenetic tree (Newick/NHX)
  &lt;k&gt;                k-mer length
  &lt;assignments.txt&gt;  assignments in generalized Kraken format

optional arguments:
  -h, --help         show this help message and exit
  -f {kraken,sam}    format of output [sam]
  -m {h1,c1}         measure: h1=hitnumber, c1=coverage [h1]
  -A                 annotate assignments
  -L                 use LCA when tie (multiple hits with the same score)
  -X                 replace k-mer matches by their LCA
  -D                 do not translate blocks from node to tax IDs
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="id3">
<h3><code class="docutils literal"><span class="pre">prophyle_analyze</span></code><a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="code text highlight-default"><div class="highlight"><pre><span></span>$ prophyle_analyze.py -h

usage: prophyle_analyze.py [-h] [-s [&#39;w&#39;, &#39;u&#39;, &#39;wl&#39;, &#39;ul&#39;]]
                           [-f [&#39;sam&#39;, &#39;bam&#39;, &#39;cram&#39;, &#39;uncompressed_bam&#39;, &#39;kraken&#39;, &#39;histo&#39;]]
                           {index_dir, tree.nw} &lt;out_prefix&gt; &lt;input_fn&gt;
                           [&lt;input_fn&gt; ...]

Program: prophyle_analyze.py

Analyze results of ProPhyle&#39;s classification.
Stats:
w: weighted assignments
u: unique assignments (ignore multiple assignments)
wl: weighted assignments, propagated to leaves
ul: unique assignments, propagated to leaves

positional arguments:
  {index_dir, tree.nw}  Index directory or phylogenetic tree
  &lt;out_prefix&gt;          Prefix for output files (the complete file names will
                        be &lt;out_prefix&gt;_rawhits.tsv for the raw hit counts
                        table and &lt;out_prefix&gt;_otu.tsv for the otu table)
  &lt;input_fn&gt;            ProPhyle output files whose format is chosen with the
                        -f option. Use &#39;-&#39; for stdin or multiple files with
                        the same format (one per sample)

optional arguments:
  -h, --help            show this help message and exit
  -s [&#39;w&#39;, &#39;u&#39;, &#39;wl&#39;, &#39;ul&#39;]
                        Statistics to use for the computation of histograms: w
                        (default) =&gt; weighted assignments; u =&gt; unique
                        assignments, non-weighted; wl =&gt; weighted assignments,
                        propagated to leaves; ul =&gt; unique assignments,
                        propagated to leaves.
  -f [&#39;sam&#39;, &#39;bam&#39;, &#39;cram&#39;, &#39;uncompressed_bam&#39;, &#39;kraken&#39;, &#39;histo&#39;]
                        Input format of assignments [auto]. If &#39;histo&#39; is
                        selected the program expects hit count histograms
                        (*_rawhits.tsv) previously computed using prophyle
                        analyze, it merges them and compute OTU table from the
                        result (assignment files are not required)
</pre></div>
</div>
</div></blockquote>
</div>
</div>
</div>


          </div>
            
  <div class="footer-relations">
    
      <div class="pull-left">
        <a class="btn btn-default" href="../tutorials/ncbi_db.html" title="previous chapter (use the left arrow)">Custom NCBI indexes</a>
      </div>
    
    </div>
    <div class="clearer"></div>
  
        </div>
        <div class="clearfix"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="../tutorials/ncbi_db.html" title="Custom NCBI indexes"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">ProPhyle  documentation</a> &#187;</li> 
      </ul>
    </div>
<script type="text/javascript">
  $("#mobile-toggle a").click(function () {
    $("#left-column").toggle();
  });
</script>
<script type="text/javascript" src="../_static/js/bootstrap.js"></script>
  <div class="footer">
    &copy; Copyright 2017, Karel Břinda, Kamil Salikhov, Simone Pignotti, Gregory Kucherov. Created using <a href="http://sphinx.pocoo.org/">Sphinx</a>.
  </div>
  </body>
</html>